<h2>Description</h2>
<p>CHM13 RNA-seq Bowtie2 default alignments and Meryl unique k-mer filtering</p>

<h2>Methods</h2>
<p>The native RNA-seq libraries were done on CHM13 cells using oligodT during library prep and sequenced for 150bp paired-end reads.</p>
<p>Reads were adapter trimmed, quality filtered before mapped to chm13v1.1 with Bowtie2 default and then samtools F1548 filtered for paired-reads only</p>
<p>Mapped reads were then filtered through unique genome-wide 21 or 51mers</p>

<h2>Display Conventions and Configuration</h2>
<!-- if there are special conventions, like colors, describe them here -->
<p></p>

<h2>Data access</h2>
<!-- Optional, links for downloading the data if there is a separate source beside the
  browser  -->
<p></p>

<h2>Release history</h2>
<ol>
  <li>CHM13v1.1 assembly (04.27.21)
</ol>

<h2>Contacts</h2>
<ul>
  <li><a href="mailto:savannah.klein@uconn.edu">Savannah Hoyt &lt;savannah.klein@uconn.edu&gt;</a>
</ul>
<h2>Credits</h2>
<p>Megan Dennis (RNAseq library prep & sequencing)</p>
<p>Colin Shew (RNAseq library prep & sequencing)</p>
<p>Savannah Hoyt (mapping & filtering)</p>
<p>Rachel O'Neill</p>
<p>Arang Rhie (Meryl software for generation of unique kmers)</p>
