* t2t-chm13-v1.0
https://hgwdev.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
https://t2t.gi.ucsc.edu/chm13/hub/hub.txt
** PATH setup
export PATH=${HOME}/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${PATH}

* rmsk
** 2020-09-25 from Mitchell Robert Vollger
/team-segdups/Assembly_analysis/Masked/chm13.draft_v1.0_plus38Y_repeatmasker.out
Includes a copy of GRCh38 chrY.

pigz chm13.draft_v1.0_plus38Y_repeatmasker.out 

# drop chrY
(../../../T2T-CHM13-hub/bin/RM2Bed.py chm13.draft_v1.0_plus38Y_repeatmasker.out.gz - | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed |tawk '$1 != "chrY"' | bedSort stdin stdout | gzip -c >rmsk.bed.gz) >&log

# switch to finishing genome before next steps
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskBed.as --twoBit=../genome/t2t-chm13-v1.0.unmasked.2bit bed9+ t2t-chm13-v1.0 rmsk.bigBed rmsk.bed.gz
ln -f rmsk.bigBed ../../../hub/t2t-chm13-v1.0/rmsk/

* trf
** 2020-09-25 from Mitchell Robert Vollger
/team-segdups/Assembly_analysis/Masked/chm13.draft_v1.0_plus38Y_trf.bed
Includes a copy of GRCh38 chrY.
pigz chm13.draft_v1.0_plus38Y_trf.bed

# drop chrY
zcat chm13.draft_v1.0_plus38Y_trf.bed.gz |tawk '$1!="chrY"' | pigz -c >trf.bed.gz

# switch to finishing genome before next steps
buildBigBed --as=../../../T2T-CHM13-hub/etc/simpleRepeat2.as bed3+ t2t-chm13-v1.0 trf.bigBed trf.bed.gz
ln -f trf.bigBed  ../../../hub/t2t-chm13-v1.0/trf/

* genome
** 2020-09-28
/assemblies/release/v1.0/chm13.draft_v1.0.fasta

pigz chm13.draft_v1.0.fasta 
faToTwoBit t2t-chm13-v1.0.unmasked.fa.gz t2t-chm13-v1.0.unmasked.2bit

twoBitMask -type=.bed t2t-chm13-v1.0.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/trf.bed.gz ) t2t-chm13-v1.0.2bit

twoBitInfo t2t-chm13-v1.0.2bit t2t-chm13-v1.0.sizes
twoBitToFa t2t-chm13-v1.0.2bit t2t-chm13-v1.0.fa

nice bgzip  --threads=64 t2t-chm13-v1.0.fa
samtools faidx t2t-chm13-v1.0.fa.gz 

blat t2t-chm13-v1.0.2bit -makeOoc=t2t-chm13-v1.0.11.ooc /dev/null /dev/null
ln -f t2t-chm13-v1.0.2bit  ../../../hub/t2t-chm13-v1.0/genome/
ln -f t2t-chm13-v1.0.fa.gz*  ../../../hub/t2t-chm13-v1.0/genome/

* sedefSegDups
** Mitchell Robert Vollger 2020-09-28
team-segdups/Assembly_analysis/SEDEF/chm13.draft_v1.0_plus38Y.SDs.bed.bb
team-segdups/Assembly_analysis/SEDEF/chm13.draft_v1.0_plus38Y.SDs.lowid.bed.bb
ln -f *.bb ../../../hub/t2t-chm13-v1.0/sedefSegDups/

* liftOffGenes
** Mitchell Robert Vollger 2020-09-28

team-segdups/Assembly_analysis/Liftoff/chm13.draft_v1.0_plus38Y.all.bb
team-segdups/Assembly_analysis/Liftoff/chm13.draft_v1.0_plus38Y.orf_only.bb
ln -f *.bb ../../../hub/t2t-chm13-v1.0/liftOffGenes

** HORs and superHORs
Tatiana Dvorkina  2020-10-05
/team-satellites/HorAnnotation/release_v1.0/HORs;
/team-satellites/HorAnnotation/release_v1.0/superHORs

cd hors
pigz *.bed
makeMd5s *.gz
buildBigBed bed9 t2t-chm13-v1.0 hors.bigBed *.gz
ln -f hors.bigBed ../../../hub/t2t-chm13-v1.0/

cd superHors
pigz *.bed
makeMd5s *.gz
buildBigBed bed9 t2t-chm13-v1.0 superHors.bigBed *.gz
ln -f superHors.bigBed ../../../hub/t2t-chm13-v1.0/

* hg38Lastz

** build alignments
cd hg38Lastz
ln -s ../../../T2T-CHM13-hub/build/t2t-chm13-v1.0/hg38Lastz.DEF DEF
runHg38Lastz fwd t2t-chm13-v1.0 >&log&
# Can't add netclass, since it relies on repeat database.
gzip -9 fwd/axtChain/noClass.net
ln fwd/axtChain/noClass.net.gz fwd/axtChain/t2t-chm13-v1.0.hg38.net.gz
ln fwd/axtChain/t2t-chm13-v1.0.hg38.* .

# swap
runHg38Lastz swap t2t-chm13-v1.0 >&log&
# script put swap in wrong place, so moved back

gzip -9 ./swap/axtChain/noClass.net 
ln ./swap/axtChain/noClass.net.gz swap/axtChain/hg38.t2t-chm13-v1.0.net.gz
ln swap/axtChain/hg38.t2t-chm13-v1.0.* .

** create over bigChains 

hgLoadChain -noBin -test no tmp/bigChain t2t-chm13-v1.0.hg38.all.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp/bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp/bigChain ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.all.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp/bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp/bigLink  ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.all.bigLink


hgLoadChain -noBin -test no tmp/bigChain t2t-chm13-v1.0.hg38.over.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp/bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp/bigChain ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.over.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp/bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp/bigLink  ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.over.bigLink

ln -f t2t-chm13-v1.0.hg38.*.big*  t2t-chm13-v1.0.hg38.*chain.gz hg38.t2t-chm13-v1.0.*.chain.gz  ../../../hub/t2t-chm13-v1.0/hg38Lastz/


* Cactus alignments
cd cactus
t2t-chm13-v1.0.aln1.hal from Marina
halRenameGenomes t2t-chm13-v1.0.aln1.hal

# build bigBed to cactus for quick browsing
cd tmp
halStats --bedSequences GRCh38 ../t2t-chm13-v1.0.aln1.hal >GRCh38.bed
split -l 1 GRCh38.bed GRCh38/chrom.

find  GRCh38 -type f | awk '{print "../getChromPairswise", $1}' >para.jobs
para create para.jobs -batch=b1
find t2t-chm13-v1.0/ -type f | xargs cat | csort -k 10,10  -k12,12n > GRCh38-t2t-chm13-v1.0.cactus.psl
pslToBigPsl GRCh38-t2t-chm13-v1.0.cactus.psl stdout | sort -k1,1 -k2,2n > GRCh38-t2t-chm13-v1.0.cactus.bigPslIn 
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as GRCh38-t2t-chm13-v1.0.cactus.bigPslIn ../../genome/t2t-chm13-v1.0.sizes ../GRCh38-t2t-chm13-v1.0.cactus.bigPsl

pslSwap GRCh38-t2t-chm13-v1.0.cactus.psl stdout | axtChain -psl -verbose=0 -scoreScheme=/scratch/data/blastz/human_chimp.v2.q -minScore=5000 -linearGap=medium stdin /hive/data/genomes/hg38/hg38.2bit ../../genome/t2t-chm13-v1.0.2bit stdout | chainSwap stdin GRCh38-t2t-chm13-v1.0.cactus.chain
chainToPsl GRCh38-t2t-chm13-v1.0.cactus.chain ../../genome/t2t-chm13-v1.0.sizes /hive/data/genomes/hg38/chrom.sizes ../../genome/t2t-chm13-v1.0.2bit /hive/data/genomes/hg38/hg38.2bit GRCh38-t2t-chm13-v1.0.cactus.chained.psl
pslToBigPsl GRCh38-t2t-chm13-v1.0.cactus.chained.psl stdout | sort -k1,1 -k2,2n > GRCh38-t2t-chm13-v1.0.cactus.chained.bigPslIn
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as GRCh38-t2t-chm13-v1.0.cactus.chained.bigPslIn  ../../genome/t2t-chm13-v1.0.sizes ../GRCh38-t2t-chm13-v1.0.cactus.chained.bigPsl

ln -f *.bigPsl *.hal ../../../hub/t2t-chm13-v1.0/cactus/

* synteny  2020-09-04
cd synteny/tmp
halStats --bedSequences t2t-chm13-v1.0 ../../cactus/t2t-chm13-v1.0.aln1.hal  > chm13.bed

create syn.tmpl:
#LOOP
../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.0.aln1.hal t2t-chm13-v1.0 $(path1) GRCh38 25000 25000 {check out exists out/25kb.$(path1).psl}
../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.0.aln1.hal t2t-chm13-v1.0 $(path1) GRCh38 1000000 50000 {check out exists out/1mb.$(path1).psl}
#ENDLOOP

gensub2 <(cut -f 1 chm13.bed) <(echo "") syn.tmpl syn.jobs

 # run with -ram=16g
para create -batch=b1 syn.jobs -ram=16g -maxQueue=100000000

# halSyntenty produced no output for chrMT, which is identical

# combine data for tracks
cat  out/25kb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >synteny.25kb.psl.gz
cat  out/1mb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >synteny.1mb.psl.gz

pslToBigPsl synteny.25kb.psl.gz 25kb.bigin
pslToBigPsl synteny.1mb.psl.gz 1mb.bigin
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 25kb.bigin ../../genome/t2t-chm13-v1.0.sizes ../synteny.25kb.bigPsl
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 1mb.bigin ../../genome/t2t-chm13-v1.0.sizes ../synteny.1mb.bigPsl

ln -f *.bigPsl ../../../hub/t2t-chm13-v1.0/synteny/

* hg002_hifiasm
Nancy Hansen
team-variants/assembly_aligns/for_browser_hub/v1/HG002_hifiasm_assemblies
mkdir -p ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/
ln HG002-v0.11.* ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/

* hg00733_hifiasm
Nancy Hansen
team-variants/assembly_aligns/for_browser_hub/v1/HG00733_hifiasm_assemblies
mkdir -p ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/hg00733_hifiasm/
ln -f HG00733-v0.11.hap* ../../../hub/t2t-chm13-v1.0/hg00733_hifiasm/


* proSeq
Savannah Hoyt
 /team-epigenetics/20200922_chm13_PROseq/STAR_alignments/
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out_BOTH.bedgraph
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Log.final.out
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_SJ.out.tab
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out_BOTH.bedgraph
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Log.final.out
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_SJ.out.tab

cd proSeq/STAR
pigz *.out *.tab *.bedgraph &
gunzip CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz &
gunzip CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz &
samtools index CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam &
samtools index CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam &

mkdir -p  ../../../hub/t2t-chm13-v1.0/proSeq/
ln -f CHM13-5*_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam* ../../../hub/t2t-chm13-v1.0/proSeq/


* wssd
Mitchell Robert Vollger
/team-segdups/WSSD_v1.0/
cd wssd
mkdir -p  ../../../hub/t2t-chm13-v1.0/wssd
ln -f *.bb  ../../../hub/t2t-chm13-v1.0/wssd

2020-10-21 - updated

* kmers
Ann Mc Cartney
 /team-satellites/Repeat_kmers/
   CHM13_Updated_k21.bed 21kmer repeat bed 
   CHM13_Updated_k21.bg - bigwig file
   t2t-chm13_k18.bed
   t2t-chm13_k18.bg

pigz CHM13_Updated_k21.bed 
buildBigBed bed4 t2t-chm13-v1.0 kmers-21.bigBed CHM13v1_k18.bed.gz 
ln CHM13_Updated_k21.bg kmers-21.bigWig

pigz t2t-chm13_k18.bed 
buildBigBed bed4 t2t-chm13-v1.0 kmers-18.bigBed t2t-chm13_k18.bed.gz
ln t2t-chm13_k18.bg kmers-18.bigWig

mkdir  ../../../hub/t2t-chm13-v1.0/kmers/

ln -f kmers-*  ../../../hub/t2t-chm13-v1.0/kmers/
## FIXMISH

* CAT
Marina
http://courtyard.gi.ucsc.edu/~mhauknes/t2t_v2/out-chm13-t2t-rebuild/

cd CAT/assemblyHub/CHM13
find . -name '*.txt' | cpio -pdum  /cluster/home/markd/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/hub/t2t-chm13-v1.0/CAT
ln -f *.bam *.bai *.bb ../../../../../hub/t2t-chm13-v1.0/CAT/


** cenSat (cenRegions/cenAnnotation
Karen
/team-satellites/cenSat_Annotation/t2t-chm13.v1.0.cenSat_annotation.bed
/team-satellites/cenSat_Annotation/t2t-chm13.v1.0.cenSat_regions.bed

pigz *.bed
# had to modify  ifles asthey had multiple tabs and sometimes space between columns
buildBigBed bed9 t2t-chm13-v1.0 cenSatAnnotation.bigBed t2t-chm13.v1.0.cenSat_annotation.bed.gz
buildBigBed bed4 t2t-chm13-v1.0 cenSatRegions.bigBed t2t-chm13.v1.0.cenSat_regions.bed.gz 
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/

* as_annotation (as_hor_annotation, as_sf_annotation)
2020-10-22 from Lev Uralsky
cd as_annotation
/team-satellites/AS_Annotation/release_v1.0/


buildBigBed bed9 t2t-chm13-v1.0 AS_HOR_Annot.bigBed t2t-chm13.release_v1.0-AS-HORs-annotation.bed.gz &
buildBigBed bed9 t2t-chm13-v1.0 AS_SF_Annot.bigBed t2t-chm13.release_v1.0-AS-SFs-annotation.bed.gz 
buildBigBed bed9 t2t-chm13-v1.0 AS_strand.bigBed t2t-chm13.release_v1.0-AS-strand.bed.gz 
ln -f AS_*.bigBed  ../../../hub/t2t-chm13-v1.0/as_annotation/




* to-do
** finish lastz



** Christopher Vollmers
@Mark Diekhans @Glennis Logsdon Here is what Mandalorion thinks the isoform sequences derived from your Iso-Seq reads are. The fasta files are straight consensus sequences generated from the subreads using PacBio consensus tools used slightly off-label. So these sequences should match CHM13 genome sequence. The psl file contains alignments of those consensus sequences to hg38 that are cleaned up, i.e. small indels are removed. One note to the output: Mandalorion will produce multiple isoforms with the same intron-chain if it thinks their ends are consistently different. Naming is Isoform_[running_number]_[number of Iso-Seq reads associated with the isoform].
chm13-isoseq--mandalorion/



** Ann Mc Cartney  1:22 PM
Hey Mark, at a hackathon for SV calls at the minute. Basically we want our tool to have a track for mappability for GRCh38. I.e regions of good and bad mapping. I'm sure UCSC has such a thing. Do you have any ideas?
https://bismap.hoffmanlab.org/

** FIMISH Ann's


** proseq
### NEED to FIGURE OUT bedgraph

Savannah Hoyt  1:07 PM

** proSeqMeryl
Savannah Hoyt
/team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_Meryl-Intersections/
PRO-seq reads aligned with Bowtie2 and then filtered with the unique 21-mers?

bedGraphToBigWig AllChrs_BOTH_chm13-5A-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort_INT_chm13-PCRfree_asm-20200922_21mer-single.bedgraph.gz ../genome/t2t-chm13-v1.0.sizes proSeqMeryl.bigWig

% Overlap between chr1 90 94 and chr1 90 94.





** cactus liftover nets with synten
** liftover comparison


** refSeqBlat
cd refSeqBlat/data
# files from 2020-05-29
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_knownrefseq_alignments.bam
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.gbff.gz
faSize -detailed GRCh38_latest_rna.fna.gz | tawk '/^N/{print $1}' >refSeq-curated.acc
faSomeRecords GRCh38_latest_rna.fna.gz refSeq-curated.acc stdout | bgzip -@ 64 -c >refSeq-curated.fa.gz

# metadata for later merge (2m16s)
gbffToTsv data/GRCh38_latest_rna.gbff.gz data/refSeq-curated.meta.tsv.gz >&log&

cd ..
blatRnaMkJobs ../genome/t2t-chm13-v1.0.2bit data/refSeq-curated.fa.gz tmp/
para make -batch=tmp/b1 tmp/para.jobs >&log&

find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >refSeq-curated.raw.gz

# filter 
faPolyASizes data/GRCh38_latest_rna.fna.gz stdout | gzip -c >data/GRCh38_latest_rna.polya.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -localNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=refSeq-curated.stats refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >refSeq-curated.psl.gz

# build bigPsl with extra columns
zcat data/refSeq-curated.meta.tsv.gz | tawk '$6!=""{print $1,$6}' > tmp/rs.cds
pslToBigPsl -cds=tmp/rs.cds -fa=data/GRCh38_latest_rna.fna.gz refSeq-curated.psl.gz stdout | sort -k1,1 -k2,2n > tmp/rs.bigin1
bigBlatPslAddColumns data/refSeq-curated.meta.tsv.gz tmp/rs.bigin1 tmp/rs.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/rs.bigin2 ../genome/t2t-chm13-v1.0.sizes refSeq-curated.bigBlatPsl
ln -f refSeq-curated.bigBlatPsl  ../../../hub/t2t-chm13-v1.0/

# 2020-09-27 discovered error with filter command, so changed above

* gencodeBlat
- use gencode V35 to match CAT, only do protein code and lncRNAs

# fix complex ids and skip PAR
zcat /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.pc_transcripts.fa.gz  /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.lncRNA_transcripts.fa.gz | gencodeFastaEdit /dev/stdin tmp/rna.fa.gz

blatRnaMkJobs ../genome/t2t-chm13-v1.0.2bit tmp/rna.fa.gz tmp

# run in parasol, MUST push all alignments, as exceeds 100000

# sort by query
find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >aligns/gencode.raw.psl.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -globalNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=aligns/gencode.stats aligns/gencode.raw.psl.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >aligns/gencode.psl.gz

hgsql hg38 -Ne 'select * from gencodeAnnotV35' | cut -f 2- |genePredToFakePsl no /dev/stdin -chromSize=/hive/data/genomes/hg38/chrom.sizes /dev/null data/gencode.cds
hgsql hg38 -e 'select ga.transcriptId id, ga.geneName as geneSym, ga.geneId, gs.geneId as hgncId, ga.geneType from gencodeAttrsV35 ga left join gencodeToGeneSymbolV35 gs on gs.transcriptId = ga.transcriptId' |gzip -c >data/gencode.meta.tsv.gz

pslToBigPsl -cds=data/gencode.cds -fa=tmp/rna.fa.gz aligns/gencode.psl.gz stdout | sort -k1,1 -k2,2n > tmp/g.bigin1

bigBlatPslAddColumns data/gencode.meta.tsv.gz tmp/g.bigin1 tmp/g.bigin2

bedToBigBed -extraIndex=name,geneSym,geneId -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/g.bigin2 ../genome/t2t-chm13-v1.0.sizes gencodeBlat.bigBlatPsl
ln -f gencodeBlat.bigBlatPsl  ../../../hub/t2t-chm13-v1.0/


* Savannah Hoyt  11:59 AM 2020-10-23
Hi Mark. I just uploaded the two separate PRO-seq duplicates to: /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_Meryl-Intersections/
Let me know if those give you any issues!

* Savannah Hoyt  11
Gotcha! I am glad they are helpful!
Yes, if you take the sorted bams (which are on Globus) and run these commands, it’ll give you POS and NEG stranded bedgraphs:
bedtools genomecov -bg -strand - -ibam CHM13_PROseq_sort.out.bam > CHM13_PROseq_sort.out_NEG.bedgraph
bedtools genomecov -bg -strand + -ibam CHM13_PROseq_sort.out.bam > CHM13_PROseq_sort.out_POS.bedgraph
The bedgraphs I added in the Bowtie2 alignments are not strand separated either. These tracks were mostly for looking in centromeric/telomeric regions as they are intersected with unique 21-mers. I’m working on generating them as separate strands, but it’s a little more intensive, so just wanted to get these two tracks up there in the interim (as long as it’s not too much trouble!).
