<h3>Description</h3>
This track represents the gene models for the T2T CHM13 assembly generated using the CAT (Compartive Annotation Toolkit) software with genes that CAT could not be mapped as well as novel paralogs, filled in from the liftOff mappings
Reference annotations are from GENCODE V35.

<h3>Display convention</h3>
This track follows the display conventions for <a href="https://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction tracks</a>. The exons for putative non-coding genes and untranslated regions are represented by relatively thin blocks, while those for coding open reading frames are thicker. Gene names are displayed in 'pack' or 'full' mode. More information about each gene can be found by clicking on the specific gene/transcript model.<br>
<br>The following color key is used: <br>
<ul>
  <li><span bgcolor="rgb(255,51,255)">Blue</span>: protein coding</li>
<li><span bgcolor="rgb(85,212,76)">Green</span>: non-coding</li>
<!-- <li><span>Purple</span>: novel predictions from PacBio Iso-Seq and LiftOff</li>  -->
<li><span bgcolor="rgb(76,85,212">Pink</span>: pseudogenes
</ul>




<br>
<h3>Methods<h3>

<h4>Comparative Annotation toolkit</h4>Genome annotation for  T2T CHM13 assembly was performed using Comparative Annotation Toolkit (CAT). CAT leverages whole-genome alignments generated by Cactus to transfer annotations from one source genome to one or more target genomes. CAT also runs AUGUSTUS in both the comparative gene prediction mode and in a single-genome mode that utilizes Iso-Seq data to predict alternative isoforms. CAT then combines all of these annotation methods into a final consensus annotation set that represents orthology relationships as well as species-specific information.
<HR>
<h2>Contacts</h2>
<ul>
  <li><a href="mailto:mhauknes@ucsc.edu">Marina Haukness &lt;mhauknes@ucsc.edu&gt;</a>
  <li><a href="mailto:ashumat2@jhmi.edu">Alaina Shumate &lt;ashumat2@jhmi.edu&gt;</a>
  <li><a href="mailto:markd@ucsc.edu">Mark Diekhans &lt;markd@ucsc.edu&gt;</a>
</ul>

<h2>Data access</h2>
<p>
Data is downloadable from Globus in <code>/team-genes/t2t-chm13-v1.0/CAT_V3/consensus_gene_set/</code>.
</p>


<h3>References</h3>
Fiddes IT, Armstrong J, Diekhans M, Nachtweide S, Kronenberg ZN, Underwood JG, Gordon D, Earl D, Keane T, Eichler EE et al. Comparative Annotation Toolkit (CAT)-simultaneous clade and personal genome annotation. Genome Res. 2018 Jul;28(7):1029-1038. PMID: 29884752; PMC: PMC6028123.
<br><br>
Stanke M, Diekhans M, Baertsch R, Haussler D. Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics. 2008 Mar 1;24(5):637-44. PMID: 18218656.
<br><br>
Stanke M, Steinkamp R, Waack S, Morgenstern B. AUGUSTUS: a web server for gene finding in eukaryotes. Nucleic Acids Res. 2004 Jul 1;32(Web Server issue):W309-12. PMID: 15215400; PMC: PMC441517.
Armstrong J, Hickey J, Diekhans M, Deran A, Fang Q, Xie D, Feng S, Stiller J, Genereux D, Johnson J, Marinescu V.D, Haussler D, Alf√∂ldi J, Lindblad-Toh K, Karlsson E, Jarvis E.D, Zhang G, Paten B. bioRxiv 730531; doi: https://doi.org/10.1101/730531
<br><br>
Shumate, A, Salzberg, S, Liftoff: an accurate gene annotation mapping tool,
DOI: 10.1101/2020.06.24.169680, https://www.biorxiv.org/content/early/2020/06/26/2020.06.24.169680
