* general stuff
** URLs
https://hgwdev.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
https://courtyard.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt

https://hgwdev.gi.ucsc.edu/~markd/t2t/hub-problem/hub.txt

** setup.sh sets variables
** globus commands:
globus endpoint search 
globus endpoint local-id
globus task list

* genome assembly download 
cd genome
globus transfer $t2t_endpoint:assemblies/drafts/t2t-chm13.20200602.fasta $my_endpoint:$(pwd -P)/t2t-chm13.20200602.fasta
bgzip t2t-chm13.20200602.fasta
makeMd5s t2t-chm13.20200602.fasta.gz 
ln t2t-chm13.20200602.fasta.gz t2tChm13_20200602.unmasked.fa.gz
faToTwoBit t2tChm13_20200602.unmasked.fa.gz t2tChm13_20200602.unmasked.2bit
* repeatmasker 
RepeatMasker version 4.1.0
Dfam_3.1
** cd rmsk
# uses modified version of browser doRepeatMasker.pl
mkdir tmp
../../../T2T-CHM13-hub/bin/doRepeatMasker.pl -stop=cat -buildDir=$(pwd)/tmp -species=mammal -unmaskedSeq=$(pwd)/../genome/t2tChm13_20200602.unmasked.2bit -useHMMER -bigClusterHub=ku -workhorse=hgwdev nodb
../../../T2T-CHM13-hub/bin/doRepeatMasker.pl -continue=cat -stop=cat -buildDir=$(pwd)/tmp -species=mammal -unmaskedSeq=$(pwd)/../genome/t2tChm13_20200602.unmasked.2bit -useHMMER -bigClusterHub=ku -workhorse=hgwdev nodb

# got:
#   ERROR: the attempt of cleaning nestedRepeats did not work+ exit 255
# just skipping and using RM2Bed
# RM2Bed also failes on bad id:
   tmp/RMPart/003/453/003453.lst.out
   174  18.9  2.1  4.8  m16_613   38473326 38473527 (4427574) +  MLT1B          LTR/ERVL-MaLR          190  390    (0)       

# drop this line
cat tmp/nodb.sorted.fa.out | fgrep -v 'm16_613   38473326 38473527'| gzip -c > t2tChm13_20200602.out.gz&
nice ../../../T2T-CHM13-hub/bin/RM2Bed.py t2tChm13_20200602.out.gz - | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin rmsk.bed >&log

bedToBigBed -type=bed9+ -as=../../../T2T-CHM13-hub/etc/rmskBed.as rmsk.bed.gz ../genome/t2tChm13_20200602.sizes rmsk.bigBed
gzip -9 rmsk.bed 
ln rmsk.bigBed ../../../hub/t2tChm13_20200602/

* trf
TRF 4.09
** cd trf
doSimpleRepeat.pl -stop=filter -buildDir=$(pwd) -unmaskedSeq=../genome/t2tChm13_20200602.unmasked.2bit -trf409=6 none  >&log&
# should have use:  -bigClusterHub=ku -workhorse=hgwdev
ln simpleRepeat.bb  ../../../hub/t2tChm13_20200602/
** results:
simpleRepeat.bb
simpleRepeat.bed.gz
trfMask.bed.gz

* mask and finish genome
twoBitMask -type=.bed t2tChm13_20200602.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/trfMask.bed.gz ) t2tChm13_20200602.2bit
twoBitInfo t2tChm13_20200602.2bit t2tChm13_20200602.sizes
twoBitToFa t2tChm13_20200602.2bit t2tChm13_20200602.fa
bgzip t2tChm13_20200602.fa
samtools faidx t2tChm13_20200602.fa.gz 
blat t2tChm13_20200602.2bit -makeOoc=t2tChm13_20200602.11.ooc /dev/null /dev/null
ln t2tChm13_20200602.2bit  ../../../hub/t2tChm13_20200602/
* cenSat1Align 
cd cenSat1Align ; mkdir tmp
doSameSpeciesLiftOver.pl cenSat1 t2tChm13_20200602 \
 -stop=net -buildDir=$(pwd)/tmp -localTmp=/dev/shm \
 -target2Bit=$(pwd)/../../../../t2t-centromere/T2T_cenSat/hub/cenSat1/cenSat1.2bit \
 -targetSizes=$(pwd)/../../../../t2t-centromere/T2T_cenSat/hub/cenSat1/cenSat1.sizes \
 -ooc=$(pwd)/../../../../t2t-centromere/T2T_cenSat/hub/cenSat1/cenSat1.11.ooc \
 -query2Bit=$(pwd)/../genome/t2tChm13_20200602.2bit \
 -querySizes=$(pwd)/../genome/t2tChm13_20200602.sizes \
 -bigClusterHub=ku -workhorse=hgwdev

* as_annotation
from Lev Uralsky
cd as_annotation
# get /team-satellites/AS_Annotation/draft_20200602/AS-HORs-draft_20200602-hmmer3.0-23042020-new-S4C20H7_8.bed.gz
makeMd5s AS-HORs-draft_20200602-hmmer3.0-23042020-new-S4C20H7_8.bed.gz
buildBigBed bed9 t2tChm13_20200602 as_annotation.bigBed AS-HORs-draft_20200602-hmmer3.0-23042020-new-S4C20H7_8.bed.gz
ln as_annotation.bigBed  ../../../hub/t2tChm13_20200602/
* merqury tracks
from Arang Rhie
cd merqury
from team-curation/merqury/merqury_*.bigWig
makeMd5s 20200602.*.bigWig
ln 20200602.read_multiplicity.bigWig  merqury_10X_reads_kmers.bigWig
ln 20200602.asm_multiplicity.bigWig   merqury_assembly_kmers.bigWig
ln 20200602.only.bigWig merqury_errors.bigWig
ln 20200602.single.bigWig merqury_marker_kmers.bigWig

ln merqury_* ../../../hub/t2tChm13_20200602/merqury/

* winnowmap_nanopore tracks
from Chirag Jain
cd winnowmap_nanopore
from team-nanopore/alignments_and_SVs/20200605/
      output.bam, output.filtered.vcf
bgzip output.filtered.vcf
makeMd5s output.*
ln output.bam winnowmap_nanopore_aligns.bam
nice samtools index -@ 32 winnowmap_nanopore_aligns.bam 

# vcf not sorted
vcf-sort output.filtered.vcf.gz |bgzip -c > winnowmap_nanopore_variants.vcf.gz
tabix -p vcf winnowmap_nanopore_variants.vcf.gz 

ln winnowmap_* ../../../hub/t2tChm13_20200602/winnowmap_nanopore/

* hor_decompositions
from Tatiana Dvorkina
/team-satellites/HorAnnotation_20200602/cen*/hordecomposition_*_20200602.bed
    hordecomposition_cen20_20200602.bed
    hordecomposition_cen3_20200602.bed
    hordecomposition_cen8_20200602.bed
    hordecomposition_cenX_20200602.bed
gzip -9 *.bed
makeMd5s *.bed.gz
buildBigBed bed9 t2tChm13_20200602 hor_decompositions.bigBed *.bed.gz
ln -f hor_decompositions.bigBed   ../../../hub/t2tChm13_20200602/
* hg38BlatAlign 
# try same species liftover going from t2tChm13_20200602 to hg38 (as target)
# since we have a subset of chroms, this should be the best direction.
# This ran for days without repeatmasking, so try it with.
cd hg38BlatAlign ; mkdir tmp
doSameSpeciesLiftOver.pl hg38 t2tChm13_20200602 \
 -stop=net -buildDir=$(pwd)/tmp -localTmp=/dev/shm \
 -target2Bit=/hive/data/genomes/hg38/hg38.2bit \
 -targetSizes=/hive/data/genomes/hg38/chrom.sizes \
 -ooc=/hive/data/genomes/hg38/jkStuff/hg38.11.ooc \
 -query2Bit=$(pwd)/../genome/t2tChm13_20200602.2bit \
 -querySizes=$(pwd)/../genome/t2tChm13_20200602.sizes \
 -bigClusterHub=ku -workhorse=hgwdev \
 -repeatMasked

* HG002vCHM13 alignments
from Sergey Aganezov <sergeyaganezovjr@gmail.com>
from /team-variants/read_aligns/bams/
tracks:
    HG002vCHM13_20200602_mm2_ONT
    HG002vCHM13_20200602_mm2_PBCCS
    HG002vCHM13_20200602_mm2_PB
    HG002vCHM13_20200602_ngmlr_PBCCS
    HG002vCHM13_20200602_ngmlr_PB
makeMd5s HG002vCHM13_20200602_*&

* to-do
** 0200611,
https://t2t-consortium.slack.com/archives/C013XAJ6U67/p1591908583197800

** Sergey Aganezov  9:35 AM
I’m now ready to share CLR/ONT/HiFi minimap2 and CLR/Hifi NGMLR alignments of HG002 reads to 20200602 assembly. I guess I just need some initial guidance on how to do said sharing. NGMLR ONT alignments are going to be ready later today/tomorrow (NGMLR is slower in general, but especially with UL ONT reads).  SVs are still being inferred/refined, and I’ll update the channel as soon as they are ready.
