HG002-X-v1.0: CHM13 + HG002-X + hg38Y  frankengenome

* genome
/public/groups/nanopore/karen/t2t-chm13/tmp/t2t-chm13.20200921.withGRCh38chrY.fasta

pigz t2t-chm13.20200921.withGRCh38chrY.fasta
faToTwoBit t2t-chm13.20200921.withGRCh38chrY.fasta.gz  HG002-X-v1.0.unmasked.2bit
ln HG002-X-v1.0.unmasked.2bit HG002-X-v1.0.2bit

twoBitInfo HG002-X-v1.0.2bit HG002-X-v1.0.sizes
tawk '{print $1,$2,"x"}' HG002-X-v1.0.sizes > HG002-X-v1.0.sizes3

ln -f HG002-X-v1.0.2bit  ../../../hub/HG002-X-v1.0/genome/

# 2021-03-23: Karen gave the wrong sequence, 
/public/groups/nanopore/karen/t2t-chm13/tmp/cutrun/HG002/chm13_HG002X_HG38Y.fasta
faToTwoBit chm13_HG002X_HG38Y.fasta.gz  HG002-X-v1.0.unmasked.2bit
ln HG002-X-v1.0.unmasked.2bit HG002-X-v1.0.2bit

twoBitInfo HG002-X-v1.0.2bit HG002-X-v1.0.sizes
tawk '{print $1,$2,"x"}' HG002-X-v1.0.sizes > HG002-X-v1.0.sizes3

ln -f HG002-X-v1.0.2bit  ../../../hub/HG002-X-v1.0/genome/

* methylation
Ariel Gershman  12:04 AM
Hi Mark, here is the methylation track for the HG002 browser /team-epigenetics/20200727_methylation/CHM13_HG002_chrX/HG002_20kb_methylation_frequency.bigwig


ln -f HG002_20kb_methylation_frequency.bigwig ../../../hub/HG002-X-v1.0/methylation/
mkdir  ../../../hub/HG002-X-v1.0/cutnrun
find CENPA IgG | cpio -pduml ../../../hub/HG002-X-v1.0/cutnrun/

* cutnrun
Savannah Hoyt
here: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/HG002_F3852_noMAPQ-filtering_bigwigs/


Ohh, the first two sets (CENPA and IgG) are in
/team-satellites/122920_CUTRUN/0222_LowSaltOnly/HG002_F3852_noMAPQ-filtering_bigwigs/

the other 3 sets (CENPB, IgG, H3K9me3) are in the FEB directory
(/team-satellites/022021_CUTRUN/bams/HG002_F3852_noMAPQ-filtering_bigwigs/).

 Since both are in /team-satellites/, but
different subdirectories should it be: 
DEC IgG:
/HG002_CutnRun-losalt-F3852/122920_CUTRUN/0222_LowSaltOnly/HG002_F3852_noMAPQ-filtering_bigwigs/filename.bigwig
FEB IgG: 
/HG002_CutnRun-losalt-F3852/022021_CUTRUN/bams/HG002_F3852_noMAPQ-filtering_bigwigs/IgG/filename.bigwig
If possible, Iâ€™d like to keep them in separate subdirectories to stay
consistent with how Karen has separated the CutnRun data by DEC vs. FEB.


 /team-satellites/122920_CUTRUN/0222_LowSaltOnly/HG002_F3852_noMAPQ-filtering_bigwigs/
-> cutnrun/dec

find dec feb | cpio -pduml ../../../hub/HG002-X-v1.0/cutnrun/
* rmskV2
from Savannah

/team-TE/Repeatmasker_polished-033121/Frankenstein_assembly/chm13-autosomes-chrM_hg002-chrX_hg38-chrY_track.bed

zcat chm13-autosomes-chrM_hg002-chrX_hg38-chrY_track.bed.gz  | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin stdout | pigz -c >rmskV2.bed.gz
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskV2Bed.as --twoBit=../genome/HG002-X-v1.0.2bit bed9+ t2t-chm13-v1.0 rmskV2.bigBed rmskV2.bed.gz
mkdir ../../../hub/HG002-X-v1.0/rmskV2/
ln -f rmskV2.bigBed ../../../hub/HG002-X-v1.0/rmskV2/

