#+STARTUP: nologdone
#+SEQ_TODO: TODO ACTIVE | DONE

* setup
export PATH=/hive/users/markd/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${PATH}

* DONE genome 
** /assemblies/release/v1.1/
# need rmsk and simpleRepeats data
faToTwoBit chm13.v1.1.fasta.gz t2t-chm13-v1.1.unmasked.2bit

faToTwoBit t2t-chm13-v1.1.unmasked.fa.gz t2t-chm13-v1.1.unmasked.2bit

twoBitMask -type=.bed t2t-chm13-v1.1.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/trf.bed.gz) t2t-chm13-v1.1.2bit

twoBitInfo t2t-chm13-v1.1.2bit t2t-chm13-v1.1.sizes
tawk '{print $1,$2,"x"}' t2t-chm13-v1.1.sizes > t2t-chm13-v1.1.sizes3

twoBitToFa t2t-chm13-v1.1.2bit t2t-chm13-v1.1.fa
nice bgzip  --threads=64 t2t-chm13-v1.1.fa
samtools faidx t2t-chm13-v1.1.fa.gz 

blat t2t-chm13-v1.1.2bit -makeOoc=t2t-chm13-v1.1.11.ooc /dev/null /dev/null
mkln t2t-chm13-v1.1.2bit t2t-chm13-v1.1.fa.gz*  ../../../../hubs/dev/t2t-chm13-v1.1/genome

* DONE trf
** 2021-06-01 Mitchell Robert Vollger
team-segdups/Assembly_analysis/Masked/chm13_v1.1_plus38Y_trf.bed
Includes a copy of GRCh38 chrY.
pigz chm13_v1.1_plus38Y_trf.bed

# drop chrY
zcat chm13_v1.1_plus38Y_trf.bed.gz |tawk '$1!="chrY"' | pigz -c >trf.bed.gz

### finish genome now
buildBigBed --as=../../../../T2T-CHM13-hub/etc/simpleRepeat2.as bed3+ t2t-chm13-v1.1 trf.bigBed trf.bed.gz
mkln trf.bigBed  ../../../../hubs/dev/t2t-chm13-v1.1/trf/

* DONE rmsk
** Savannah Hoyt
/team-TE/Repeatmasker_polished-033121/chm13v1.1_polish-050321/
chm13v1.1_RM-polish-0502121_trackformat.bed
editRepeatMaskerBed chm13v1.1_RM-polish-0502121_trackformat.bed | sort -k1,1 -k2,2n | pigz -c >rmsk.bed.gz

### finish genome now

# switch to finishing genome before next steps
buildBigBed  --as=../../../../T2T-CHM13-hub/etc/rmskV2Bed.as --twoBit=../genome/t2t-chm13-v1.1.unmasked.2bit bed9+ t2t-chm13-v1.1 rmsk.bigBed rmsk.bed.gz
mkln rmsk.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/rmsk/

* DONE sedefSegDups
** 2021-06-01 Mitchell Robert Vollger
team-segdups/Assembly_analysis/SEDEF/chm13_v1.1_plus38Y.SDs.bed.bb
team-segdups/Assembly_analysis/SEDEF/chm13_v1.1_plus38Y.SDs.lowid.bed.bb
mkln *.bb ../../../../hubs/dev/t2t-chm13-v1.1/sedefSegDups/
* DONE gc5Base
hgGcPercent -noLoad -wigOut -doGaps -win=5 -file=stdout -verbose=0 NODB ../genome/t2t-chm13-v1.1.2bit  | pigz -c > gc5Base.wigVarStep.gz
wigToBigWig gc5Base.wigVarStep.gz ../genome/t2t-chm13-v1.1.sizes gc5Base.bigWig
mkln gc5Base.bigWig ../../../../hubs/dev/t2t-chm13-v1.1/gc5Base
* DONE cpgIsland
twoBitToFa ../genome/t2t-chm13-v1.1.2bit stdout | /hive/data/staging/data/cpgIslandExt/cpglh /dev/stdin >cpgIslands.cpglh
./my-doMakeBed.csh 
mkln t2t-chm13-v1.1.cpgIslandExt.bb  ../../../../hubs/dev/t2t-chm13-v1.1/cpgIsland

*  v1.0_to_v1.1.html
Sergey Koren
@Arang Rhie made one and itâ€™s on the chm13 GitHub page though it will fail to lift over the borders of rdna so needs spot checking

* LASTZ IMPORTANT
 liftover should not have fix
Justin Zook @MarkD - as @Nancy Hansen and I were looking into variants that don't liftover concordantly between GRCh38 and CHM13, we noticed an anomaly in the liftover from CHM13 to GRCh38, where a multi-Mbp region of CHM13 seems to have its primary alignment to chr8_KZ208915v1_fix instead of to chr8.  This region contains TNKS, MSRA, MFHAS1, BLK, GATA4, and a few more genes. Do you have any idea why this might be happening? It's probably too late to change this for our paper, but might be good to investigate and resolve at some point.

* grch38PriLastz

** Lastz parameters (from human/chimp browser settings)
v1.03.52

BLASTZ_O=600    --gap=600,150
BLASTZ_E=150
BLASTZ_M=254     --masking=254
BLASTZ_T=2       --seed=12of19 --notransition
BLASTZ_Y=15000 --ydrop==15000
BLASTZ_K=4500  --hspthresh=4500

BLASTZ_Q --score=
    A    C    G    T
    90 -330 -236 -356
  -330  100 -318 -236
  -236 -318  100 -330
  -356 -236 -330   90

** create grch38 files with only primary assembly
cd grch38PriLastz/tmp/grch38
fgrep -v _ /hive/data/genomes/hg38/chrom.sizes >grch38.sizes
twoBitToFa -seqList=<(cut -f 1 grch38.sizes) /hive/data/genomes/hg38/hg38.2bit stdout | faToTwoBit stdin grch38.2bit

** build alignments
cd grch38PriLastz
ln -s ../../../../../T2T-CHM13-hub/dev/build/t2t-chm13-v1.1/grch38PriLastz.DEF DEF
runLastz fwd t2t-chm13-v1.1 >&fwd.log&

# after complete
runLastz swap t2t-chm13-v1.1 >&swap.log&

# Can't add netclass, since it relies on repeat database.
pigz -9 axtChain/noClass.net
ln axtChain/noClass.net.gz ../t2t-chm13-v1.1.grch38.net.gz
ln axtChain/t2t-chm13-v1.1.grch38.* ..

pigz -9 ./swap/axtChain/noClass.net 
ln ./swap/axtChain/noClass.net.gz ../grch38.t2t-chm13-v1.1.net.gz
ln swap/axtChain/grch38.t2t-chm13-v1.1.* ..

** create over bigChains 
cd tmp
hgLoadChain -noBin -test no ignore ../t2t-chm13-v1.1.grch38.all.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > chain.bigBedIn
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab chain.bigBedIn  ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.grch38.all.bigChain
tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > link.bigBedIn
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab link.bigBedIn ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.grch38.all.bigLink

hgLoadChain -noBin -test no ignore ../t2t-chm13-v1.1.grch38.over.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > chain.bigBedIn
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab chain.bigBedIn  ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.grch38.over.bigChain
tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > link.bigBedIn
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab link.bigBedIn ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.grch38.over.bigLink

** link to hub
# make source chains available
mkln *.gz *.big* ../../../../hubs/dev/t2t-chm13-v1.1/grch38PriLastz/
mkln *.chain.gz ../../../../hubs/dev/t2t-chm13-v1.1/downloads/

* proseq
Savannah: /team-epigenetics/PROseq-RNAseq_chm13v1.1/MappedToCHM13v1.1/
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1_POS.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1_POS.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_POS.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-21mer-chm13v1.1_POS.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_meryl-51mer-chm13v1.1_POS.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_NEG.bigwig
PROseq_BT2-default/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5A_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-21mer-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_meryl-51mer-chm13v1.1_POS.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_NEG.bigwig
PROseq_BT2-k100/CHM13-5B_proseq_cutadapt-q20-m20_bt2-vs-dM_bt2-k100-chm13v1.1_POS.bigwig

mkln PROseq_BT2-default/* ../../../../hubs/dev/t2t-chm13-v1.1/proseq/PROseq_BT2-default/
mkln PROseq_BT2-k100/* ../../../../hubs/dev/t2t-chm13-v1.1/proseq/PROseq_BT2-k100

* rnaseq
Savannah: /team-epigenetics/PROseq-RNAseq_chm13v1.1/MappedToCHM13v1.1/
RNAseq_BT2-default/CHM13_S182_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548.bigwig
RNAseq_BT2-default/CHM13_S182_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548_meryl-21mer-chm13v1.1.bigwig
RNAseq_BT2-default/CHM13_S182_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548_meryl-51mer-chm13v1.1.bigwig
RNAseq_BT2-default/CHM13_S183_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548.bigwig
RNAseq_BT2-default/CHM13_S183_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548_meryl-21mer-chm13v1.1.bigwig
RNAseq_BT2-default/CHM13_S183_rnaseq_cutadapt-q20-m100_bt2-chm13v1.1_F1548_meryl-51mer-chm13v1.1.bigwig

mkln RNAseq_BT2-default/*.bigwig ../../../../hubs/dev/t2t-chm13-v1.1/rnaseq/RNAseq_BT2-default/

* cactus 2021-08-23
from marina
courtyard:/nanopore/marina/t2t_v1.1/t2tChm13.v1.1.hal
mv t2tChm13.v1.1.hal t2t-chm13-v1.1.hal
# edit hal to use t2tChm13.v1.1 and hg38 names
halRenameGenomes t2t-chm13-v1.1.hal renames.tab
mkln t2t-chm13-v1.1.hal ../../../../hubs/dev/t2t-chm13-v1.1/cactus/

cd tmp
halStats --bedSequences hg38 ../t2t-chm13-v1.1.hal >hg38.bed
split -l 1 hg38.bed hg38/chrom.

find  hg38 -type f | awk '{print "../getChromPairswise", $1}' >para.jobs
para create para.jobs -batch=b1
find t2t-chm13-v1.1/ -type f | xargs cat | csort -k 10,10  -k12,12n > t2t-chm13-v1.1.hg38.cactus.psl
pslToBigPsl t2t-chm13-v1.1.hg38.cactus.psl stdout | sort -k1,1 -k2,2n > t2t-chm13-v1.1.hg38.cactus.bigPslIn 
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as t2t-chm13-v1.1.hg38.cactus.bigPslIn ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.hg38.cactus.bigPsl

pslSwap t2t-chm13-v1.1.hg38.cactus.psl stdout | axtChain -psl -verbose=0 -scoreScheme=/scratch/data/blastz/human_chimp.v2.q -minScore=5000 -linearGap=medium stdin /hive/data/genomes/hg38/hg38.2bit ../../genome/t2t-chm13-v1.1.2bit stdout | chainSwap stdin t2t-chm13-v1.1.hg38.cactus.chain
chainToPsl t2t-chm13-v1.1.hg38.cactus.chain ../../genome/t2t-chm13-v1.1.sizes /hive/data/genomes/hg38/chrom.sizes ../../genome/t2t-chm13-v1.1.2bit /hive/data/genomes/hg38/hg38.2bit t2t-chm13-v1.1.hg38.cactus.chained.psl
pslToBigPsl t2t-chm13-v1.1.hg38.cactus.chained.psl stdout | sort -k1,1 -k2,2n > t2t-chm13-v1.1.hg38.cactus.chained.bigPslIn
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as t2t-chm13-v1.1.hg38.cactus.chained.bigPslIn  ../../genome/t2t-chm13-v1.1.sizes ../t2t-chm13-v1.1.hg38.cactus.chained.bigPsl

# get lifted 1.0 PSL from Marina
/nanopore/marina/t2t_v1.1/chain/t2t-chm13-v1.1.liftover.GRCh38.cactus.psl
pslToBigPsl t2t-chm13-v1.1.liftover.GRCh38.cactus.psl stdout | sort -k1,1 -k2,2n > tmp/t2t-chm13-v1.1.liftover.hg38.cactus.bigPslIn
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as tmp/t2t-chm13-v1.1.liftover.hg38.cactus.bigPslIn  ../genome/t2t-chm13-v1.1.sizes t2t-chm13-v1.1.liftover.hg38.cactus.bigPsl

mkln *.bigPsl ../../../../hubs/dev/t2t-chm13-v1.1/cactus/

# cactus chains
# NOT DONE
# added halChainChrom to build


* issues  Assembly/Validation
Arang Rhie
https://raw.githubusercontent.com/marbl/CHM13-issues/main/v1.1_issues.bed
https://raw.githubusercontent.com/marbl/CHM13-issues/main/description.html

buildBigBed bed9  t2t-chm13-v1.1 issues.bigBed v1.1_issues.bed 
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/issues* synteny  2021-08-23
cd synteny/tmp
halStats --bedSequences t2t-chm13-v1.1 ../../cactus/t2t-chm13-v1.1.hal  > t2t.bed

create syn.tmpl:
#LOOP
../../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.1.hal t2t-chm13-v1.1 $(path1) hg38 25000 25000 {check out exists out/25kb.$(path1).psl}
../../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.1.hal t2t-chm13-v1.1 $(path1) hg38 1000000 50000 {check out exists out/1mb.$(path1).psl}
#ENDLOOP

gensub2 <(cut -f 1 t2t.bed) <(echo "") syn.tmpl syn.jobs

 # run with -ram=16g
para create -batch=b1 syn.jobs -ram=16g -maxQueue=100000000

# halSyntenty produced no output for chrMT, which is identical

# combine data for tracks
cat out/25kb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |pigz -9 >synteny.25kb.psl.gz
cat out/1mb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |pigz -9 >synteny.1mb.psl.gz

pslToBigPsl synteny.25kb.psl.gz 25kb.bigin
pslToBigPsl synteny.1mb.psl.gz 1mb.bigin
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 25kb.bigin ../../genome/t2t-chm13-v1.1.sizes ../synteny.25kb.bigPsl
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 1mb.bigin ../../genome/t2t-chm13-v1.1.sizes ../synteny.1mb.bigPsl

cd ..
mkln *.bigPsl ../../../../hubs/dev/t2t-chm13-v1.1/synteny
* uwLiftoff 2021-09-28
Mitchell 
team-segdups/Assembly_analysis/Liftoff/chm13_v1.1_plus38Y.all.bb
team-segdups/Assembly_analysis/Liftoff/chm13_v1.1_plus38Y.orf_only.bb

mkln chm13_v1.1_plus38Y.* ../../../../hubs/dev/t2t-chm13-v1.1/uwLiftoff/

* compositeRepeats 2021-09-29
Savannah files set by slack

buildBigBed bed4 t2t-chm13-v1.1 compositeRepeats.bigBed Composite_track_CHM13v1.1.bed.gz 

buildBigBed bed4 t2t-chm13-v1.1 compositeRepeats.bigBed
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/compositeRepeats

# make available for download for the paper
http://t2t.gi.ucsc.edu/chm13/dev/t2t-chm13-v1.1/downloads/t2t-chm13-v1.1.composite-repeats-singletons-arrays.bed.gz
ln -f Composite_track_CHM13v1.1.bed.gz  ../../../../hubs/dev/t2t-chm13-v1.1/downloads/t2t-chm13-v1.1.composite-repeats-singletons-arrays.bed.gz

* newSatellitesMonomersArrays 2021-09-29
Savannah files set by slack

buildBigBed bed4 t2t-chm13-v1.1 newSatellitesMonomersArrays.bigBed NewSatellites_track_CHM13v1.1.bed.gz 

mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/newSatellitesMonomersArrays

# make available for download for the paper
http://t2t.gi.ucsc.edu/chm13/dev/t2t-chm13-v1.1/downloads/t2t-chm13-v1.1.new-satellites-monomers-arrays.bed.gz
ln Composite_track_CHM13v1.1.bed.gz  ../../../../hubs/dev/t2t-chm13-v1.1/downloads/t2t-chm13-v1.1.new-satellites-monomers-arrays.bed.gz


* as_annotation 2021-09-29
Lev 
The three annotation tracks for CHM13 v1.1 with the description files are
on Globus at /team-satellites/AS_Annotation/CHM13_v1.1/

pigz *.bed


buildBigBed bed9 t2t-chm13-v1.1 AS_HOR_Annot.bigBed t2t-chm13.v1.1-AS-HORs-annotation.170921.bed.gz &
buildBigBed bed9 t2t-chm13-v1.1 AS_SF_Annot.bigBed t2t-chm13.v1.1-AS-SFs.290621.bed.gz &
buildBigBed bed9 t2t-chm13-v1.1 AS_strand.bigBed t2t-chm13.v1.1-AS-strand.170921.bed.gz &

mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/as_annotation/

* cytobands 2021-10-01
from @Nick Altemose via slack

chm13_v1.1.cytoBandIdeo_v1.1.txt
buildBigBed bed4 --as=${HOME}/kent/src/hg/lib/cytoBand.as   t2t-chm13-v1.1  cytoBandIdeo.bigBed chm13_v1.1.cytoBandIdeo_v1.1.txt
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.1/cytobands/
* CAT_V4
lifted from V1.0
http://courtyard.gi.ucsc.edu/~mhauknes/t2t_v1.1/liftover/chm13.draft_v1.1.gene_annotation.v4.bb
mkln assemblyHub/CHM13/*  ../../../../hubs/dev//t2t-chm13-v1.1/CAT_V4

* ACTIVE gencodeBlat
- use gencode V35 to match CAT, only do protein code and lncRNAs

### NOTE for next time, the RNA FASTAs don't include sequences on unplaced/randoms, also didn't include
### pseudogenes

# only use sequences from primary assembly
hgsql hg38 -Ne 'select name from wgEncodeGencodeCompV35 where length(chrom) <=5' >data/gencode.pri.ids
hgsql hg38 -Ne 'select name from wgEncodeGencodePseudoGeneV35 where length(chrom) <=5' >>data/gencode.pri.ids

# fix complex ids and skip PAR
zcat /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.pc_transcripts.fa.gz  /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.lncRNA_transcripts.fa.gz | gencodeFastaEdit --ids=data/gencode.pri.ids /dev/stdin data/rna.fa.gz

blatRnaMkJobs ../genome/t2t-chm13-v1.1.2bit data/rna.fa.gz tmp

# run in parasol, MUST push all alignments, as exceeds 100000
para push -maxPush=500000 -batch=tmp/b2 tmp/para.jobs 

# sort by query
find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >aligns/gencode.raw.psl.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -globalNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=aligns/gencode.stats aligns/gencode.raw.psl.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >aligns/gencode.psl.gzp

hgsql hg38 -Ne 'select * from gencodeAnnotV35' | cut -f 2- |genePredToFakePsl no /dev/stdin -chromSize=/hive/data/genomes/hg38/chrom.sizes /dev/null data/gencode.cds
hgsql hg38 -e 'select ga.transcriptId id, ga.geneName as geneSym, ga.geneId, gs.geneId as hgncId, ga.geneType from gencodeAttrsV35 ga left join gencodeToGeneSymbolV35 gs on gs.transcriptId = ga.transcriptId' |gzip -c >data/gencode.meta.tsv.gz

pslToBigPsl -cds=data/gencode.cds -fa=data/rna.fa.gz aligns/gencode.psl.gz stdout | sort -k1,1 -k2,2n > tmp/g.bigin1

bigBlatPslAddColumns data/gencode.meta.tsv.gz tmp/g.bigin1 tmp/g.bigin2

bedToBigBed -extraIndex=name,geneSym,geneId -type=bed12+17 -tab -as=../../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/g.bigin2 ../genome/t2t-chm13-v1.1.sizes gencodeBlat.bigBlatPsl
mkln gencodeBlat.bigBlatPsl ../../../../hubs/dev/t2t-chm13-v1.1/gencodeBlat

* TODO refSeqBlat
For Ann & Arang to compare RefSeq of  20200904 and v1.1

cd refSeqBlat/data
# files dated from 2020-08-19  but same as when download 2020-05-29
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.gbff.gz
faSize -detailed GRCh38_latest_rna.fna.gz | tawk '/^N/{print $1}' >refSeq-curated.acc
faSomeRecords GRCh38_latest_rna.fna.gz refSeq-curated.acc stdout | pigz -c >refSeq-curated.fa.gz
faSize -detailed data/refSeq-curated.fa.gz >data/refSeq-curated.sizes

# metadata for later merge
gbffToTsv GRCh38_latest_rna.gbff.gz refSeq-curated.meta.tsv.gz >&log&
faPolyASizes GRCh38_latest_rna.fna.gz stdout | pigz -c >GRCh38_latest_rna.polya.gz
# CDS too
gbToFaRa /dev/null /dev/null GRCh38_latest_rna.ra /dev/null GRCh38_latest_rna.gbff.gz 
tawko '$1=="acc"{acc=$2} $1=="cds"{print acc,$2}' GRCh38_latest_rna.ra  >refSeq-curated.cds
pigz data/refSeq-curated.cds data/GRCh38_latest_rna.ra 

cd ..
blatRnaMkJobs ../genome/t2t-chm13-v1.1.2bit data/refSeq-curated.fa.gz tmp/
para make -batch=tmp/b1 tmp/para.jobs >&log&

find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >refSeq-curated.raw.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -localNearBest=0.001 -minId=0.92 -minCover=0.75 -statsOut=refSeq-curated.stats refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >refSeq-curated.psl.gz

# build bigPsl with extra columns
zcat data/refSeq-curated.meta.tsv.gz | tawk '$6!=""{print $1,$6}' > tmp/rs.cds
pslToBigPsl -cds=tmp/rs.cds -fa=data/GRCh38_latest_rna.fna.gz refSeq-curated.psl.gz stdout | sort -k1,1 -k2,2n > tmp/rs.bigin1
bigBlatPslAddColumns data/refSeq-curated.meta.tsv.gz tmp/rs.bigin1 tmp/rs.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/rs.bigin2 ../genome/t2t-chm13-v1.1.sizes refSeq-curated.bigBlatPsl
ln -f refSeq-curated.bigBlatPsl  ../../../hub/t2t-chm13-v1.1/

# build a BAM for Ann
tawk 'BEGIN{print "@HD", "VN:1.6","SO:coordinate"} {print "@SQ", "SN:"$1, "LN:"$2}' ../genome/t2t-chm13-v1.1.sizes >header.sam
(cat header.sam; zcat refSeq-curated.psl.gz |psl2sam.pl) | samtools view -b >refSeq-curated.bam

# get genePred to look at frame
mrnaToGene -quiet -genePredExt -cdsFile=data/refSeq-curated.cds.gz -insertMergeSize=0 refSeq-curated.psl.gz refSeq-curated.gp
pigz *.gp


