<h2>Description</h2>
<p>Precision Run-On (PRO) Sequencing Bowtie2 alignment anchored to single-copy 21-mers found in assembly-filtered PCR-free WGS reads</p>

<h2>Methods</h2>
<p>The PRO-seq experiment was done on CHM13 cells and sequenced for 75bp single-ended reads.</p>
<p>Reads were trimmed and filtered and then aligned to the T2T CHM13 v1.0 assembly (released on 20200922) using bowtie2/2.3.5.1 with the following parameters:</p>
<p>-N 0</p>
<p>-k 100</p>
<p>Following alignment, the reads were anchored to single-copy 21-mers found in assembly-filtered PCR-free WGS reads generated with meryl/1.2 using GenomeBrowser/20180626 overlapSelect with the following parameters:</p>
<p>-overlapBases=21</p>

<h2>Display Conventions and Configuration</h2>
<!-- if there are special conventions, like colors, describe them here -->
<p></p>

<h2>Data access</h2>
<!-- Optional, links for downloading the data if there is a separate source beside the
  browser  -->
<p></p>

<h2>Release history</h2>
<ol>
  <li>2020-09-22: CHM13 v1.0 release (chm13.draft_v1.0.fasta)
</ol>

<h2>Contacts</h2>
<ul>
  <li><a href="mailto:rachel.oneill@uconn.edu">Rachel O'Neill &lt;rachel.oneill@uconn.edu&gt;</a>
  <li><a href="mailto:savannah.klein@uconn.edu">Savannah Hoyt &lt;savannah.klein@uconn.edu&gt;</a>
</ul>
<h2>Credits</h2>
<p>Rachel O'Neill</p>
<p>Savannah Hoyt</p>
<p>Arang Rhie (Meryl software)</p>