* t2t-chm13-v1.0
https://hgwdev.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
https://t2t.gi.ucsc.edu/chm13/hub/hub.txt
** PATH setup
export PATH=${HOME}/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${PATH}

* rmsk
** 2020-09-25 from Mitchell Robert Vollger
/team-segdups/Assembly_analysis/Masked/chm13.draft_v1.0_plus38Y_repeatmasker.out
Includes a copy of GRCh38 chrY.

pigz chm13.draft_v1.0_plus38Y_repeatmasker.out 

# drop chrY
(../../../T2T-CHM13-hub/bin/RM2Bed.py chm13.draft_v1.0_plus38Y_repeatmasker.out.gz - | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed |tawk '$1 != "chrY"' | bedSort stdin stdout | gzip -c >rmsk.bed.gz) >&log

# switch to finishing genome before next steps
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskBed.as --twoBit=../genome/t2t-chm13-v1.0.unmasked.2bit bed9+ t2t-chm13-v1.0 rmsk.bigBed rmsk.bed.gz
ln -f rmsk.bigBed ../../../hub/t2t-chm13-v1.0/rmsk/

* trf
** 2020-09-25 from Mitchell Robert Vollger
/team-segdups/Assembly_analysis/Masked/chm13.draft_v1.0_plus38Y_trf.bed
Includes a copy of GRCh38 chrY.
pigz chm13.draft_v1.0_plus38Y_trf.bed

# drop chrY
zcat chm13.draft_v1.0_plus38Y_trf.bed.gz |tawk '$1!="chrY"' | pigz -c >trf.bed.gz

# switch to finishing genome before next steps
buildBigBed --as=../../../T2T-CHM13-hub/etc/simpleRepeat2.as bed3+ t2t-chm13-v1.0 trf.bigBed trf.bed.gz
ln -f trf.bigBed  ../../../hub/t2t-chm13-v1.0/trf/

* genome
** 2020-09-28
/assemblies/release/v1.0/chm13.draft_v1.0.fasta

pigz chm13.draft_v1.0.fasta 
faToTwoBit t2t-chm13-v1.0.unmasked.fa.gz t2t-chm13-v1.0.unmasked.2bit

twoBitMask -type=.bed t2t-chm13-v1.0.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/trf.bed.gz ) t2t-chm13-v1.0.2bit

twoBitInfo t2t-chm13-v1.0.2bit t2t-chm13-v1.0.sizes
twoBitToFa t2t-chm13-v1.0.2bit t2t-chm13-v1.0.fa
tawk '{print $1,$2,"x"}' t2t-chm13-v1.0.sizes > t2t-chm13-v1.0.sizes3

nice bgzip  --threads=64 t2t-chm13-v1.0.fa
samtools faidx t2t-chm13-v1.0.fa.gz 

blat t2t-chm13-v1.0.2bit -makeOoc=t2t-chm13-v1.0.11.ooc /dev/null /dev/null
ln -f t2t-chm13-v1.0.2bit  ../../../hub/t2t-chm13-v1.0/genome/
ln -f t2t-chm13-v1.0.fa.gz*  ../../../hub/t2t-chm13-v1.0/genome/

* sedefSegDups
** Mitchell Robert Vollger 2020-09-28
team-segdups/Assembly_analysis/SEDEF/chm13.draft_v1.0_plus38Y.SDs.bed.bb
team-segdups/Assembly_analysis/SEDEF/chm13.draft_v1.0_plus38Y.SDs.lowid.bed.bb
ln -f *.bb ../../../hub/t2t-chm13-v1.0/sedefSegDups/

* liftOffGenes
** Mitchell Robert Vollger 2020-09-28

team-segdups/Assembly_analysis/Liftoff/chm13.draft_v1.0_plus38Y.all.bb
team-segdups/Assembly_analysis/Liftoff/chm13.draft_v1.0_plus38Y.orf_only.bb
ln -f *.bb ../../../hub/t2t-chm13-v1.0/liftOffGenes

* HORs and superHORs
Tatiana Dvorkina  2020-10-05
/team-satellites/HorAnnotation/release_v1.0/HORs;
/team-satellites/HorAnnotation/release_v1.0/superHORs

cd hors
pigz *.bed
makeMd5s *.gz
buildBigBed bed9 t2t-chm13-v1.0 hors.bigBed *.gz
ln -f hors.bigBed ../../../hub/t2t-chm13-v1.0/

cd superHors
pigz *.bed
makeMd5s *.gz
buildBigBed bed9 t2t-chm13-v1.0 superHors.bigBed *.gz
ln -f superHors.bigBed ../../../hub/t2t-chm13-v1.0/

* hg38Lastz
Lastz parameters (from human/chimp browser settings)
v1.03.52

BLASTZ_O=600    --gap=600,150
BLASTZ_E=150
BLASTZ_M=254     --masking=254
BLASTZ_T=2       --seed=12of19 --notransition
BLASTZ_Y=15000 --ydrop==15000
BLASTZ_K=4500  --hspthresh=4500

BLASTZ_Q --score=
    A    C    G    T
    90 -330 -236 -356
  -330  100 -318 -236
  -236 -318  100 -330
  -356 -236 -330   90


** build alignments
cd hg38Lastz
ln -s ../../../T2T-CHM13-hub/build/t2t-chm13-v1.0/hg38Lastz.DEF DEF
runHg38Lastz fwd t2t-chm13-v1.0 >&log&
# Can't add netclass, since it relies on repeat database.
gzip -9 fwd/axtChain/noClass.net
ln fwd/axtChain/noClass.net.gz fwd/axtChain/t2t-chm13-v1.0.hg38.net.gz
ln fwd/axtChain/t2t-chm13-v1.0.hg38.* .

# swap
runHg38Lastz swap t2t-chm13-v1.0 >&log&
# script put swap in wrong place, so moved back

gzip -9 ./swap/axtChain/noClass.net 
ln ./swap/axtChain/noClass.net.gz swap/axtChain/hg38.t2t-chm13-v1.0.net.gz
ln swap/axtChain/hg38.t2t-chm13-v1.0.* .

** create over bigChains 

hgLoadChain -noBin -test no tmp/bigChain t2t-chm13-v1.0.hg38.all.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp/bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp/bigChain ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.all.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp/bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp/bigLink  ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.all.bigLink


hgLoadChain -noBin -test no tmp/bigChain t2t-chm13-v1.0.hg38.over.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp/bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp/bigChain ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.over.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp/bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp/bigLink  ../genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0.hg38.over.bigLink

ln -f t2t-chm13-v1.0.hg38.*.big*  t2t-chm13-v1.0.hg38.*chain.gz hg38.t2t-chm13-v1.0.*.chain.gz  ../../../hub/t2t-chm13-v1.0/hg38Lastz/

# make source chains available
ln -f *.all.chain.gz  ../../../hub/t2t-chm13-v1.0/hg38Lastz/

* Cactus alignments
cd cactus
t2t-chm13-v1.0.aln1.hal from Marina
halRenameGenomes t2t-chm13-v1.0.aln1.hal

# build bigBed to cactus for quick browsing
# use target.query.xxx file nameing convention to match lastz
cd tmp
halStats --bedSequences GRCh38 ../t2t-chm13-v1.0.aln1.hal >GRCh38.bed
split -l 1 GRCh38.bed GRCh38/chrom.

find  GRCh38 -type f | awk '{print "../getChromPairswise", $1}' >para.jobs
para create para.jobs -batch=b1
find t2t-chm13-v1.0/ -type f | xargs cat | csort -k 10,10  -k12,12n > t2t-chm13-v1.0.GRCh38.cactus.psl
pslToBigPsl t2t-chm13-v1.0.GRCh38.cactus.psl stdout | sort -k1,1 -k2,2n > t2t-chm13-v1.0.GRCh38.cactus.bigPslIn 
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as t2t-chm13-v1.0.GRCh38.cactus.bigPslIn ../../genome/t2t-chm13-v1.0.sizes ../t2t-chm13-v1.0.GRCh38.cactus.bigPsl

pslSwap t2t-chm13-v1.0.GRCh38.cactus.psl stdout | axtChain -psl -verbose=0 -scoreScheme=/scratch/data/blastz/human_chimp.v2.q -minScore=5000 -linearGap=medium stdin /hive/data/genomes/hg38/hg38.2bit ../../genome/t2t-chm13-v1.0.2bit stdout | chainSwap stdin t2t-chm13-v1.0.GRCh38.cactus.chain
chainToPsl t2t-chm13-v1.0.GRCh38.cactus.chain ../../genome/t2t-chm13-v1.0.sizes /hive/data/genomes/hg38/chrom.sizes ../../genome/t2t-chm13-v1.0.2bit /hive/data/genomes/hg38/hg38.2bit t2t-chm13-v1.0.GRCh38.cactus.chained.psl
pslToBigPsl t2t-chm13-v1.0.GRCh38.cactus.chained.psl stdout | sort -k1,1 -k2,2n > t2t-chm13-v1.0.GRCh38.cactus.chained.bigPslIn
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as t2t-chm13-v1.0.GRCh38.cactus.chained.bigPslIn  ../../genome/t2t-chm13-v1.0.sizes ../t2t-chm13-v1.0.GRCh38.cactus.chained.bigPsl

# make syntenic chains
chainNet 
netSyntenic 

ln -f *.bigPsl *.hal ../../../hub/t2t-chm13-v1.0/cactus/

* synteny  2020-09-04
cd synteny/tmp
halStats --bedSequences t2t-chm13-v1.0 ../../cactus/t2t-chm13-v1.0.aln1.hal  > chm13.bed

create syn.tmpl:
#LOOP
../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.0.aln1.hal t2t-chm13-v1.0 $(path1) GRCh38 25000 25000 {check out exists out/25kb.$(path1).psl}
../../../../T2T-CHM13-hub/bin/halSyntenyRun  ../../cactus/t2t-chm13-v1.0.aln1.hal t2t-chm13-v1.0 $(path1) GRCh38 1000000 50000 {check out exists out/1mb.$(path1).psl}
#ENDLOOP

gensub2 <(cut -f 1 chm13.bed) <(echo "") syn.tmpl syn.jobs

 # run with -ram=16g
para create -batch=b1 syn.jobs -ram=16g -maxQueue=100000000

# halSyntenty produced no output for chrMT, which is identical

# combine data for tracks
cat  out/25kb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >synteny.25kb.psl.gz
cat  out/1mb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >synteny.1mb.psl.gz

pslToBigPsl synteny.25kb.psl.gz 25kb.bigin
pslToBigPsl synteny.1mb.psl.gz 1mb.bigin
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 25kb.bigin ../../genome/t2t-chm13-v1.0.sizes ../synteny.25kb.bigPsl
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 1mb.bigin ../../genome/t2t-chm13-v1.0.sizes ../synteny.1mb.bigPsl

ln -f *.bigPsl ../../../hub/t2t-chm13-v1.0/synteny/

* hg002_hifiasm
Nancy Hansen
team-variants/assembly_aligns/for_browser_hub/v1/HG002_hifiasm_assemblies
mkdir -p ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/
ln HG002-v0.11.* ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/

* hg00733_hifiasm
Nancy Hansen
team-variants/assembly_aligns/for_browser_hub/v1/HG00733_hifiasm_assemblies
mkdir -p ../../../hub/t2t-chm13-v1.0/hg002_hifiasm/hg00733_hifiasm/
ln -f HG00733-v0.11.hap* ../../../hub/t2t-chm13-v1.0/hg00733_hifiasm/

* proSeq
Savannah Hoyt
 /team-epigenetics/20200922_chm13_PROseq/STAR_alignments/
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out_BOTH.bedgraph
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Log.final.out
CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_SJ.out.tab
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out_BOTH.bedgraph
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Log.final.out
CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_SJ.out.tab

cd proSeq/STAR
pigz *.out *.tab *.bedgraph &
gunzip CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz &
gunzip CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam.gz &
samtools index CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam &
samtools index CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam &

mkdir -p  ../../../hub/t2t-chm13-v1.0/proSeq/
ln -f CHM13-5*_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam* ../../../hub/t2t-chm13-v1.0/proSeq/

# how previous wiggle tracks were generate:
bedtools genomecov -bg -strand - -ibam CHM13_PROseq_sort.out.bam > CHM13_PROseq_sort.out_NEG.bedgraph
bedtools genomecov -bg -strand + -ibam CHM13_PROseq_sort.out.bam > CHM13_PROseq_sort.out_POS.bedgraph

* proSeqMeryl
Savannah Hoyt
/team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_Meryl-Intersections/

# redone with overlap select
AllChrs_BOTH_IlluminaPCRfree_20200922-assembly_21mers_single_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed.gz
AllChrs_BOTH_IlluminaPCRfree_20200922-assembly_21mers_single_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed.gz

buildBigBed bed6 t2t-chm13-v1.0 proSeqMerylA.bigBed AllChrs_BOTH_IlluminaPCRfree_20200922-assembly_21mers_single_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed.gz &
buildBigBed bed6 t2t-chm13-v1.0 proSeqMerylB.bigBed AllChrs_BOTH_IlluminaPCRfree_20200922-assembly_21mers_single_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed.gz &

mkdir -p  ../../../hub/t2t-chm13-v1.0/proSeqMeryl/
ln -f proSeqMeryl*.bigBed  ../../../hub/t2t-chm13-v1.0/proSeqMeryl/

2020-12-16 51-MER  Savannah Hoyt
PRO-seq/Meryl 51-mer overlaps to the browser (both replicates & track description: /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_51mer-Meryl-Intersections/), as well as the Meryl 51-mers themselves (which I believe are here from Arang: /team-curation/marker_assisted/20200921_v1/).
IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed
IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed

buildBigBed bed6 t2t-chm13-v1.0 proSeqMeryl51A.bigBed  IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922.bed.gz &

* wssd
Mitchell Robert Vollger
/team-segdups/WSSD_v1.0/
cd wssd
mkdir -p  ../../../hub/t2t-chm13-v1.0/wssd
ln -f *.bb  ../../../hub/t2t-chm13-v1.0/wssd

2020-10-21 - updated

* kmersTandem
Ann Mc Cartney
 /team-satellites/Repeat_kmers/
   CHM13_Updated_k21.bed 21kmer repeat bed 
   CHM13_Updated_k21.bg - bedgraph file

pigz CHM13_Updated_k21.bg  
pigz CHM13_Updated_k21.bed 
buildBigBed bed4 t2t-chm13-v1.0 kmers-21-tandem.bigBed CHM13v1_k18.bed.gz 
zcat CHM13_Updated_k21.bg.gz | tail -n +2 | TMDIR=/dev/shm/ sort -k1,1 -k2,2n >tmp.bedgraph
bedGraphToBigWig tmp.bedgraph ../genome/t2t-chm13-v1.0.sizes kmers-21-tandem.bigWig

mkdir  ../../../hub/t2t-chm13-v1.0/kmers/
ln -f kmers-*  ../../../hub/t2t-chm13-v1.0/kmers/

* kmersUnique 
/team-genes/chm13v1_Tracks/20200921.single.bed
pigz 20200921.single.bed
buildBigBed bed3 t2t-chm13-v1.0 kmers-21-uniq.bigBed 20200921.single.bed.gz
ln -f kmers-*  ../../../hub/t2t-chm13-v1.0/kmers/

# 2020-12-16 add 51 base unique kmers
/team-curation/marker_assisted/20200921_v1/chm13.v1.single.k51.mrg.bed
 chm13.v1.single.k51.mrg.bed 
 buildBigBed bed3 t2t-chm13-v1.0 kmers-51-uniq.bigBed chm13.v1.single.k51.mrg.bed.gz

* CAT
Marina
http://courtyard.gi.ucsc.edu/~mhauknes/t2t_v2/out-chm13-t2t-rebuild/

cd CAT/assemblyHub/CHM13
find . -name '*.txt' | cpio -pdum  /cluster/home/markd/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/hub/t2t-chm13-v1.0/CAT
ln -f *.bam *.bai *.bb ../../../../../hub/t2t-chm13-v1.0/CAT/

cp consensus_gene_set/* ../../../hub/t2t-chm13-v1.0/CAT/consensus_gene_set/
pigz ../../../hub/t2t-chm13-v1.0/CAT/consensus_gene_set/*


** cenSat (cenRegions/cenAnnotation
Karen
/team-satellites/cenSat_Annotation/t2t-chm13.v1.0.cenSat_annotation.bed
/team-satellites/cenSat_Annotation/t2t-chm13.v1.0.cenSat_regions.bed

pigz *.bed
# had to modify  ifles asthey had multiple tabs and sometimes space between columns
buildBigBed bed9 t2t-chm13-v1.0 cenSatAnnotation.bigBed t2t-chm13.v1.0.cenSat_annotation.bed.gz
buildBigBed bed4 t2t-chm13-v1.0 cenSatRegions.bigBed t2t-chm13.v1.0.cenSat_regions.bed.gz 
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/ls 

# 2020-11-10 with t2t-chm13.v1.0.1110.cenSat_annotation.bed
buildBigBed bed9 t2t-chm13-v1.0 cenSatAnnotation.bigBed  t2t-chm13.v1.0.1110.cenSat_annotation.bed.gz 

* as_annotation (as_hor_annotation, as_sf_annotation)
2020-10-22 from Lev Uralsky
cd as_annotation
/team-satellites/AS_Annotation/release_v1.0/


buildBigBed bed9 t2t-chm13-v1.0 AS_HOR_Annot.bigBed t2t-chm13.release_v1.0-AS-HORs-annotation.bed.gz &
buildBigBed bed9 t2t-chm13-v1.0 AS_SF_Annot.bigBed t2t-chm13.release_v1.0-AS-SFs-annotation.bed.gz 
buildBigBed bed9 t2t-chm13-v1.0 AS_strand.bigBed t2t-chm13.release_v1.0-AS-strand.bed.gz 
ln -f AS_*.bigBed  ../../../hub/t2t-chm13-v1.0/as_annotation/



* gencodeBlat
### NOTE for next time, the RNA FASTAs don't include sequences on unplaced/randoms, also didn't include
### pseudogenes

- use gencode V35 to match CAT, only do protein code and lncRNAs

# fix complex ids and skip PAR
zcat /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.pc_transcripts.fa.gz  /hive/data/genomes/hg38/bed/gencodeV35/import/release/release_35/gencode.v35.lncRNA_transcripts.fa.gz | gencodeFastaEdit /dev/stdin data/rna.fa.gz

blatRnaMkJobs ../genome/t2t-chm13-v1.0.2bit data/rna.fa.gz tmp

# run in parasol, MUST push all alignments, as exceeds 100000

# sort by query
find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >aligns/gencode.raw.psl.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -globalNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=aligns/gencode.stats aligns/gencode.raw.psl.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >aligns/gencode.psl.gz

hgsql hg38 -Ne 'select * from gencodeAnnotV35' | cut -f 2- |genePredToFakePsl no /dev/stdin -chromSize=/hive/data/genomes/hg38/chrom.sizes /dev/null data/gencode.cds
hgsql hg38 -e 'select ga.transcriptId id, ga.geneName as geneSym, ga.geneId, gs.geneId as hgncId, ga.geneType from gencodeAttrsV35 ga left join gencodeToGeneSymbolV35 gs on gs.transcriptId = ga.transcriptId' |gzip -c >data/gencode.meta.tsv.gz

pslToBigPsl -cds=data/gencode.cds -fa=data/rna.fa.gz aligns/gencode.psl.gz stdout | sort -k1,1 -k2,2n > tmp/g.bigin1

bigBlatPslAddColumns data/gencode.meta.tsv.gz tmp/g.bigin1 tmp/g.bigin2

bedToBigBed -extraIndex=name,geneSym,geneId -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/g.bigin2 ../genome/t2t-chm13-v1.0.sizes gencodeBlat.bigBlatPsl
ln -f gencodeBlat.bigBlatPsl  ../../../hub/t2t-chm13-v1.0/

* GC percent 
cd gc5Base
hgGcPercent -noLoad -wigOut -doGaps -win=5 -file=stdout -verbose=0 NODB ../genome/t2t-chm13-v1.0.2bit  | gzip -c > gc5Base.wigVarStep.gz
wigToBigWig gc5Base.wigVarStep.gz  ../genome/t2t-chm13-v1.0.sizes gc5Base.bigWig
ln -f gc5Base.bigWig  ../../../hub/t2t-chm13-v1.0/

* gap track
cd gap
../../../T2T-CHM13-hub/bin/faToGapBed ../genome/t2t-chm13-v1.0.fa.gz gap.bed
buildBigBed bed3 t2t-chm13-v1.0 gap.bigBed gap.bed
ln -f gap.bigBed  ../../../hub/t2t-chm13-v1.0
** 2021-05-03
# edit file to add name rDNA
buildBigBed bed4 t2t-chm13-v1.0 gap.bigBed gap.bed
ln -f gap.bigBed  ../../../hub/t2t-chm13-v1.0

* crispr
cd crispr

# added -forHub option
~/compbio/nanopore/projs/t2t-chm13/kent/src/hg/utils/automation/doCrispr.pl -continue XXX  -stop XXX  -forHub -buildDir $(pwd)/build -twoBit $(pwd)/../genome/t2t-chm13-v1.0.2bit t2t-chm13-v1.0 >&log &


for each of
    indexFa
    ranges
    guides - run failed jobs with 16gb, then 32gb, then finish runGuides.bash
    specScoreJobList
    specScores
    effScores
    offTargets
    effScores


# couldn't modify out-of-sync with git scripts in /hive/data/outside/crisprTrack/scripts, so
# had to make make these links


  ln -s /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/crispr/build/genomes/t2t-chm13-v1.0 /hive/data/outside/crisprTrack/crispor/genomes/
  ln /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/genome/t2t-chm13-v1.0.2bit /hive/data/genomes/t2t-chm13-v1.0/
  ln /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/genome/t2t-chm13-v1.0.sizes /hive/data/genomes/t2t-chm13-v1.0/chrom.sizes
  ln build/indexFa/t2t-chm13-v1.0.fa* /data/outside/crisprTrack/crispor/genomes/t2t-chm13-v1.0//

 #  
ln build/crispr.bb .
ln build/crisprDetails.tab .

ln -f crispr.bb crisprDetails.tab ../../../hub/t2t-chm13-v1.0/crispr/

** methyFreq 
Ariel Gershman
methylation bigwig on globus /team-epigenetics/20200727_methylation/v1.0_methylation/methylation_frequency.bigwig would it be possible for you to put that on the t2t browser? Thank you

ln -f methylation_frequency.bigwig   ../../../hub/t2t-chm13-v1.0/methyFreq.bigWig

* salmonQuant
2021-01-12 from Colin Shew <cshewr@ucdavis.edu>
team-epigenetics/chm13_rnaseq/salmon_quant
ln -f chm13.CATv1.salmonQuant.bb  ../../../hub/t2t-chm13-v1.0/

* salmonQuantV4
2021-05-21 from Colin Shew <cshewr@ucdavis.edu>
/team-epigenetics/chm13_rnaseq/salmon_quant/CHM13.CATv4.salmonQuant.bb
/team-epigenetics/chm13_rnaseq/salmon_quant/CHM13.CATv4.salmonQuant.html
ln -f CHM13.CATv4.salmonQuant.bb  ../../../hub/t2t-chm13-v1.0/

* refSeqBlat
For Ann & Arang to compare RefSeq of  20200904 and v1.0

cd refSeqBlat/data
# files dated from 2020-08-19  but same as when download 2020-05-29
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.gbff.gz
faSize -detailed GRCh38_latest_rna.fna.gz | tawk '/^N/{print $1}' >refSeq-curated.acc
faSomeRecords GRCh38_latest_rna.fna.gz refSeq-curated.acc stdout | pigz -c >refSeq-curated.fa.gz
faSize -detailed data/refSeq-curated.fa.gz >data/refSeq-curated.sizes

# metadata for later merge
gbffToTsv GRCh38_latest_rna.gbff.gz refSeq-curated.meta.tsv.gz >&log&
faPolyASizes GRCh38_latest_rna.fna.gz stdout | pigz -c >GRCh38_latest_rna.polya.gz
# CDS too
gbToFaRa /dev/null /dev/null GRCh38_latest_rna.ra /dev/null GRCh38_latest_rna.gbff.gz 
tawko '$1=="acc"{acc=$2} $1=="cds"{print acc,$2}' GRCh38_latest_rna.ra  >refSeq-curated.cds
pigz data/refSeq-curated.cds data/GRCh38_latest_rna.ra 

cd ..
blatRnaMkJobs ../genome/t2t-chm13-v1.0.2bit data/refSeq-curated.fa.gz tmp/
para make -batch=tmp/b1 tmp/para.jobs >&log&

find tmp/psls -type f | xargs cat | sort -k 10,10 | pigz -c >refSeq-curated.raw.gz

pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -localNearBest=0.001 -minId=0.92 -minCover=0.75 -statsOut=refSeq-curated.stats refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >refSeq-curated.psl.gz

# build bigPsl with extra columns
zcat data/refSeq-curated.meta.tsv.gz | tawk '$6!=""{print $1,$6}' > tmp/rs.cds
pslToBigPsl -cds=tmp/rs.cds -fa=data/GRCh38_latest_rna.fna.gz refSeq-curated.psl.gz stdout | sort -k1,1 -k2,2n > tmp/rs.bigin1
bigBlatPslAddColumns data/refSeq-curated.meta.tsv.gz tmp/rs.bigin1 tmp/rs.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/rs.bigin2 ../genome/t2t-chm13-v1.0.sizes refSeq-curated.bigBlatPsl
ln -f refSeq-curated.bigBlatPsl  ../../../hub/t2t-chm13-v1.0/

# build a BAM for Ann
tawk 'BEGIN{print "@HD", "VN:1.6","SO:coordinate"} {print "@SQ", "SN:"$1, "LN:"$2}' ../genome/t2t-chm13-v1.0.sizes >header.sam
(cat header.sam; zcat refSeq-curated.psl.gz |psl2sam.pl) | samtools view -b >refSeq-curated.bam

# get genePred to look at frame
mrnaToGene -quiet -genePredExt -cdsFile=data/refSeq-curated.cds.gz -insertMergeSize=0 refSeq-curated.psl.gz refSeq-curated.gp
pigz *.gp


* illuminaMarkerDeserts
Ann Mc Cartney
/team-genes/bed_tracks/illumina_marker_deserts.bed

buildBigBed bed4 t2t-chm13-v1.0 illuminaMarkerDeserts.bigBed illumina_marker_deserts.bed.gz 

* proSeqV2
2021-02-15
####PROseq and RNAseq data on UCSC T2T v1.0 browser
##As of FEB 15th, 2021
##Mapped & filtered by Savannah Hoyt (UCONN)

** PROseq (50bp SE):

*** PROseq_BT2 Mapped only (Bowtie2 -N 0 -k 100): /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_only/
	#Name in browser: PROseq_BT2
	#PROseq_BT2-N0-k100_chm13-v1.0_trackDescription.html
	#CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sorted.bigwig
	#CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sorted.bigwig

*** PROseq_BT2_Meryl-21mers Mapped and filtered through Meryl unique 21mers (overlapSelect -overlapBases=21): /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_21mer-Meryl-Intersections/
	#Name in browser: PROseq_BT2_Meryl-21mers
	#PROseq_BT2-N0-k100_Meryl-uniq-21mer_PCRfree-assembly-chm13-v1.0_trackDescription.html
	#IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort.bigwig
	#IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort.bigwig

*** PROseq_BT2_Meryl-51mers Mapped and filtered through Meryl unique 51mers (overlapSelect -overlapBases=51): /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2-aligned_51mer-Meryl-Intersections/
	#Name in browser: PROseq_BT2_Meryl-51mers
	#PROseq_BT2-N0-k100_Meryl-uniq-51mer_PCRfree-assembly-chm13-v1.0_trackDescription.html
	#IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort.bigwig
	#IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort.bigwig

** RNAseq (150bp PE):

*** RNAseq_BT2 Mapped only (Bowtie2 -N 0 -k 100; samtools F1548): /team-epigenetics/20200922-v1.0_chm13_PROseq/CHM13_RNAseq_analysis/Bowtie2-aligned_F1548/
	#Name in browser: RNAseq_BT2
	#RNAseq_BT2-N0-k100_chm13-v1.0_trackDescription.html
	#CHM13_1_S182_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig
	#CHM13_2_S183_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig

*** RNAseq_BT2_Meryl-21mers Mapped and filtered through Meryl unique 21mers (overlapSelect -overlapBases=21): /team-epigenetics/20200922-v1.0_chm13_PROseq/CHM13_RNAseq_analysis/Bowtie2-aligned_21mer-Meryl-Intersections/
	#Name in browser: RNAseq_BT2_Meryl-21mers
	#RNAseq_BT2-N0-k100_Meryl-uniq-21mer_PCRfree-assembly-chm13-v1.0_trackDescription.html
	#IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig
	#IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig

*** RNAseq_BT2_Meryl-51mers Mapped and filtered through Meryl unique 51mers (overlapSelect -overlapBases=51): /team-epigenetics/20200922-v1.0_chm13_PROseq/CHM13_RNAseq_analysis/Bowtie2-aligned_51mer-Meryl-Intersections/
	#Name in browser: RNAseq_BT2_Meryl-51mers
	#RNAseq_BT2-N0-k100_Meryl-uniq-51mer_PCRfree-assembly-chm13-v1.0_trackDescription.html
	#IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig
	#IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F1548_sort.bigwig

** install
find . -name '*.bigwig' | cpio -pduml  ../../../hub/t2t-chm13-v1.0/

** additional files 2021-03-28

* WSSD_20210216 (wssd update)
Mitchell Robert Vollger
I have updated the two WSSD annotation sets and added an additional one. Could you update the browser for these three new / modified tracks?
/team-segdups/WSSD_20210216/

# move directories up a level and link subset to save disk space
ln wssd_sub/* wssd/
find . | cpio -pduml  ../../../hub/t2t-chm13-v1.0/

** cenSatV2 (cenRegions/cenAnnotation) update v2.021621
directly from Karen

pigz *.bed
buildBigBed bed9 t2t-chm13-v1.0 cenSatAnnotation.bigBed  t2t_cenAnnotation.v2.021721.bed.gz 
buildBigBed bed9 t2t-chm13-v1.0 cenSatRegions.bigBed t2t_cenRegions.v2.021621.bed.gz
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/

# 2021-02-19 updated
buildBigBed bed9 t2t-chm13-v1.0 cenSatAnnotation.bigBed t2t_cenAnnotation.v2.021921.bed.bz2 
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/

* cutnrun 2021-02-17
Savannah
Track description html for ALL files
** Location of files on Globus (one directory per sample): T2T_CutnRun_MappedandFiltered.txt
####Cut & Run data on UCSC T2T v1.0 browser
##As of FEB 17th, 2021
##Data from Gina Caldas: 150bp Paired-end
##Mapped & samtools filtered (F1548) by Karen Miga: bwa mem -k 50 -c 1000000
##Unique k-mer filtered by Savannah Hoyt: overlapSelect -overlapBases=[size_of_kmer]
##Unique 21mers and 51mers from Meryl software (Arang Rhie)
##Unique 100mers (Michael Sauria)

###CENPA: /team-satellites/122920_CUTRUN/bams_F1548/CENPA/
	#CHM13_CA_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_CA_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_CA_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_CA_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
	
###CENPB: /team-satellites/122920_CUTRUN/bams_F1548/CENPB/
	#CHM13_CB_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_CB_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_CB_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_CB_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
	
###CENPC: /team-satellites/122920_CUTRUN/bams_F1548/CENPC/
	#CHM13_CC_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_CC_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_CC_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_CC_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
	
###H3K4me2: /team-satellites/122920_CUTRUN/bams_F1548/H3K4me2/
	#CHM13_H3K4me2_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_H3K4me2_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_H3K4me2_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_H3K4me2_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
	
###H3K9me3: /team-satellites/122920_CUTRUN/bams_F1548/H3K9me3/
	#CHM13_H3K9me3_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_H3K9me3_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_H3K9me3_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_H3K9me3_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
	
###IgG: /team-satellites/122920_CUTRUN/bams_F1548/IgG/
	#CHM13_IgG_cutnrun_hi_losalt_trimmed_q20_2.chm13.draft_v1.0.F1548.sorted.bigwig
	#CHM13_IgG_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
	#CHM13_IgG_cutnrun_hisalt_losalt.q20.F1548.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
	#CHM13_IgG_cutnrun_hisalt_losalt.q20.F1548.over.chm13v1_unique_100mers_sort.bigwig
** The edited trackDb file (which I listed all files and edited the track names, short labels and long labels; I did not adjust any of the other settings listed though!)

find . | cpio -pduml  ../../../hub/t2t-chm13-v1.0/cutnrun

** replaced 2021-02-22
find CENP* H3K* IgG | cpio -pduml  ../../../hub/t2t-chm13-v1.0/cutnrun

* low-coverage
Arang Rhie team-curation/tracks/v1/issues

buildBigBed bed9 t2t-chm13-v1.0 low_coverage.bigBed low_coverage.bed.gz
buildBigBed bed9 t2t-chm13-v1.0 low_coverage.30k.bigBed low_coverage.30k.bed.gz
buildBigBed bed9 t2t-chm13-v1.0 low_coverage.100k.bigBed low_coverage.100k.bed.gz
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/low_coverage/
** REPLACED BY low-coverage2
* jaspar:
** https://github.com/wassermanlab/JASPAR-UCSC-tracks

python3 -m virtualenv venv
pyenv venv
pip install Biopython==1.73
pip install NumPy pyfaidx tqdm
cd venv/
wget https://raw.githubusercontent.com/wassermanlab/JASPAR-UCSC-tracks/master/install-pwmscan.sh
bash -x install-pwmscan.sh 
git clone git@github.com:wassermanlab/JASPAR-UCSC-tracks.git
ln -s ../bin JASPAR-UCSC-tracks/
zcat ../../../genome/t2t-chm13-v1.0.fa.gz >../t2t-chm13-v1.0.fa

# hack code to check errors
nice JASPAR-UCSC-tracks/scan_sequence.py --fasta-file ../t2t-chm13-v1.0.fa --dummy-dir=/data/tmp --profiles-dir JASPAR-UCSC-tracks/profiles/ --output-dir ../beds --threads 32 --latest --taxon vertebrates >&log

(nice ./JASPAR-UCSC-tracks/scans2bigBed -d /data/tmp -c ../../../genome/t2t-chm13-v1.0.sizes -i ../beds -o ../jaspar.bigBed -t 32) >&log2&

ln -f jaspar.bigBed ../../../hub/t2t-chm13-v1.0/jaspar/


* cutnrun-losalt/ 2021-03-01
Savannah
** ####CHM13 Cut & Run data on Globus
##As of MARCH 1st, 2021
##Low salt only data from Gina Caldas/Nick Altemose: 150bp Paired-end
##Mapped & samtools filtered (F3852) by Karen Miga: bwa mem -k 50 -c 1000000
##Unique k-mer filtered by Savannah Hoyt: overlapSelect -overlapBases=[size_of_kmer]
##Unique 21mers and 51mers from Meryl software (Arang Rhie)
##Unique 100mers (Michael Sauria)

#######Sequenced in DEC 2020:

###CENPA: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPA/
CHM13_CA_cutnrun_losalt_trimmed_q20_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CA_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CA_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CA_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###CENPB: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPB/
CHM13_CB_cutnrun_losalt_trimmed_q20_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CB_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CB_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CB_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###CENPC: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPC/
CHM13_CC_cutnrun_losalt_trimmed_q20_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CC_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CC_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CC_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###H3K4me2: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/H3K4me2/
CHM13_H3K4me2_cutnrun_losalt_trimmed_q20_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_H3K4me2_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_H3K4me2_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_H3K4me2_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###H3K9me3: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/H3K9me3/
CHM13_H3K9me3_cutnrun_losalt_trimmed_q20_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_H3K9me3_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_H3K9me3_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_H3K9me3_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###IgG: /team-satellites/122920_CUTRUN/0222_LowSaltOnly/CHM13_F3852_noMAPQ-filtering_bigwigs/IgG/
CHM13_IgG_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig
CHM13_IgG_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_IgG_cutnrun_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_IgG_cutnrun_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

#######Sequenced in FEB 2021:

###CENPA: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPA/
CHM13_CA_ADGCCR06-E_S5_L001.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CA_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CA_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CA_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###CENPB: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPB/
CHM13_CB_ADGCCR09-A_S96_L002_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CB_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CB_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CB_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###CENPC: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/CENPC/
CHM13_CC_ADGCCR09-B_S97_L002_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_CC_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_CC_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_CC_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

###H3K27me3: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/H3K27me3/
CHM13_K27me3_ADGCCR09-E_S100_L002_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_H3K27me3_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_H3K27me3_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_H3K27me3_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

####H3K9me3: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/H3K9me3/
CHM13_K9Me3_ADGCCR09-D_S99_L002_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_H3K9me3_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_H3K9me3_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_H3K9me3_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

####IgG: /team-satellites/022021_CUTRUN/bams/CHM13_F3852_noMAPQ-filtering_bigwigs/IgG/
CHM13_IgG_ADGCCR09-C_S98_L002_2.chm13.draft_v1.0F3852_sort.bigwig
CHM13_IgG_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_21mers_single_mrg_meryl_sort.bigwig
CHM13_IgG_cutnrun-202021_losalt.F3852.over.IlluminaPCRfree_v1.0-assembly_51mers_single_mrg_meryl_sort.bigwig
CHM13_IgG_cutnrun-202021_losalt.F3852.over.chm13v1_unique_100mers_sort.bigwig

find CENP* H3K* IgG | cpio -pduml  ../../../hub/t2t-chm13-v1.0/cutnrun-losalt/


* proSeq_F3852
Savannah Hoyt 2021-03-04
** ####PROseq and RNAseq data on UCSC T2T v1.0 browser
##As of MARCH 2, 2021
##Bowtie2 mapped, Samtools filtered (F3852), unique 21-mer and 51-mer filtered (Savannah Hoyt)
##PRO-seq experiment and sequencing done by Savannah Hoyt
##RNA-seq experiment (with oligoDT) and sequencing done by Megan Dennis and Colin Shew

###PROseq (75bp SE):

##Mapped only (Bowtie2 -N 0 -k 100; F3852): /team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/PROseq/not-kmer_filtered/
	#Bed files:
		CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
		CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
	
##Mapped and filtered through Meryl unique 21mers (overlapSelect -overlapBases=21): /team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/PROseq/kmer_filtered/
	#Bed files:
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		
##Mapped and filtered through Meryl unique 51mers (overlapSelect -overlapBases=51):/team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/PROseq/kmer_filtered/
	#Bed files:
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig

###RNAseq (150bp PE):

##Mapped only (Bowtie2 -N 0 -k 100; samtools F3852): /team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/RNAseq/not-kmer_filtered/
	#Bed files:
		CHM13_1_S182_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
		CHM13_2_S183_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		CHM13_1_S182_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		CHM13_2_S183_RNAseq_cutadapt-m100_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
	
##Mapped and filtered through Meryl unique 21mers (overlapSelect -overlapBases=21): /team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/RNAseq/kmer_filtered/
	#Bed files:
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852.bed
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		IlluminaPCRfree_20200922-assembly_21mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
	
##Mapped and filtered through Meryl unique 51mers (overlapSelect -overlapBases=51): /team-epigenetics/20200922-v1.0_chm13_PROseq/PROseq_RNAseq_Bowtie2_F3852/RNAseq/kmer_filtered/
	#Bed files:
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852.bed
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852.bed
	#Bigwigs:
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_1_S182_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
		IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13_2_S183_RNAseq_cutadapt-m100_bt2-N0-k100-chm13-20200922_F3852_sort.bigwig
	
** build
find PROseq RNAseq | cpio -pduml  ../../../hub/t2t-chm13-v1.0/proSeq_F3852

* liftOff fill in
Alaina Shumate  
here is a new file with missing genes/extra copies that liftoff picked up.  This should be the set that can be added to the CAT annotations to form the union. Nothing in this set overlaps any other CAT annotations
ftp://ftp.ccb.jhu.edu//pub/ashumate/CHM13_Liftoff/v2/CHM13.liftoff.unique.with.phase.gff3
cd CAT/liftOffFill
gff3ToGenePred CHM13.liftoff.unique.with.phase.gff3.gz CHM13.liftoff.unique.gp


* cutnrun-kmer-enrichment
Pragya Sidhwani 2021-03-08
globus:/team-satellites/ref_free_enriched_kmers/
ln *.bw ../../../hub/t2t-chm13-v1.0/cutnrun-kmer-enrichment/


* encode_macs2 2021-03-11
Michael Sauria
I've put all of the files and the track descriptions on the T2T Finishing Workshop endpoint at /team-epigenetics/031121_chm13_encode_macs2/. If you can create a composite track (encode_analysis.html) of 3 composite tracks (encode_pileups,html, encode_macs2_peaks.html, and encode_macs2_liftover_peaks.html),

makeTrackDb generate most of trackDb
find . -name '*.bb' -o -name '*.bw' | cpio -pduml ../../../hub/t2t-chm13-v1.0/encode_macs2

* liftOffV2
Alaina Shumate  
/team-genes/t2t-chm13-v1.0/Liftoff
2021-03-19 updated to correct missing exons on some transcripts

gff3ToGenePred -attrsOut=liftOffV2.attrs CHM13.liftoff.all.v2.gff3.gz  liftOffV2.gp
zcat liftOffV2.attrs.gz | tawk '$2=="gene_name"{print $1, $3,$3}' >liftOffV2.geneNames.tab

genePredToBigGenePred -geneNames=liftOffV2.geneNames.tab liftOffV2.gp.gz  stdout | sort -k1,1 -k2,2n > tmp.bgpInput
wget -nv https://genome.ucsc.edu/goldenpath/help/examples/bigGenePred.as
bedToBigBed -extraIndex=name,name2,geneName -type=bed12+8 -tab -as=bigGenePred.as tmp.bgpInput ../genome/t2t-chm13-v1.0.sizes liftOffV2.bigGenePred
ln -f liftOffV2.bigGenePred  ../../../hub/t2t-chm13-v1.0/liftOffV2/

gencode v35

* proSeqSplitStrands
** 2021-03-17 Savannah
/team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_split-strands/

ln -f *.bigwig ../../../hub/t2t-chm13-v1.0/proSeqSplitStrands/

## 2021-03-27 - 4 more files
CHM13-5A-B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort_NEG.bigwig
CHM13-5A-B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort_POS.bigwig
IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A-B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort_NEG.bigwig
IlluminaPCRfree_20200922-assembly_51mers_single_mrg_meryl.over.CHM13-5A-B_proseq_cutadapt-2_bt2-vs-filtered_bt2-N0-k100-chm13-20200922_sort_POS.bigwig

ln -f *.bigwig ../../../hub/t2t-chm13-v1.0/proSeqSplitStrands/

* proSeqSplitStrands2
** 2021-04-19 update from Savannah
All 16 files are here: /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_k100/PROseq/
They are strand-separated so I included the color and altColor in the trackDB.
ln -f *.bigwig ../../../hub/t2t-chm13-v1.0/proSeqSplitStrands2/

# replaces previous proSeqSplitStrands 

* rnaSeqKmerFilt2
** 2021-04-19 update from Savannah

/team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_k100/RNAseq/

ln -f *.bigwig ../../../hub/t2t-chm13-v1.0/rnaSeqKmerFilt2/
# replaces proSeq_F3852.trackDb.txt

* rmskV2 new repeat master
2021-03-19 Savannah
/team-TE/Repeatmasker_polished/chm13-v1_Repeatmasker_polish_031921_track_v2.bed
# note, has C form - strand, that must be edits

(zcat chm13-v1_Repeatmasker_polish_031921_track_v2.bed.gz |../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin stdout | gzip -c >rmskV2.bed.gz )>&log

# checked for warnings, but did not update known defaults or colors

buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskV2Bed.as --twoBit=../genome/t2t-chm13-v1.0.unmasked.2bit bed9+ t2t-chm13-v1.0 rmskV2.bigBed rmskV2.bed.gz
ln -f rmskV2.bigBed ../../../hub/t2t-chm13-v1.0/rmskV2/


# BSR is different, fixed in code.
V2 name                     BSR             
Class of repeat             Beta
Subclass repeat             Satellite

V1 name                     BSR/Beta
Class of repeat             Satellite
Family of repeat            unknown

** 2021-03-31
update from Savannah that fixed a bug
team-TE/Repeatmasker_polished-033121/chm13v1_polish-033121_track.bed 

# one bad record with spaces, fix with sed
zcat chm13v1_polish-033121_track.bed.gz | sed -e 's/chr10   27055808 /chr10\t27055808/' | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin stdout | pigz -c >rmskV2.bed.gz
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskV2Bed.as --twoBit=../genome/t2t-chm13-v1.0.unmasked.2bit bed9+ t2t-chm13-v1.0 rmskV2.bigBed rmskV2.bed.gz
ln -f rmskV2.bigBed ../../../hub/t2t-chm13-v1.0/rmskV2/


* CAT_V2
Marina

gencode v35
ok, that worked, everything is updated! all the relevant files here
http://courtyard.gi.ucsc.edu/~mhauknes/t2t_v2/out-chm13-draft_v1-full-isoseq-chm13-only/analysis/consensus_v2/

new big bed, I think this worked better, let me know if not: http://courtyard.gi.ucsc.edu/~mhauknes/t2t_v2/out-chm13-draft_v1-full-isoseq-chm13-only/analysis/consensus_v2/CHM13.combined.v2.bb

# copy over and munge entire hub
# a bunch of editing of trackDb files
mkdir ../../../hub/t2t-chm13-v1.0/CAT_V2
ln -f assemblyHub/CHM13/*.bam assemblyHub/CHM13/*.bai assemblyHub/CHM13/*.bb ../../../hub/t2t-chm13-v1.0/CAT_V2
find consensus_gene_set | cpio -pduml ../../../hub/t2t-chm13-v1.0/CAT_V2/

** tmp browser for liftoff regions
bedToHtmlDir -title 'liftoff fill' -hub https://t2t.gi.ucsc.edu/chm13/hub/hub.txt -sort-by-loc t2t-chm13-v1.0 CHM13.liftoff.unique.v2.bed  ~/public_html/t2t/tmp/liftOffFill/
** create data set for gencode V35 source transcripts

* CAT_V2.1
# fix to liftoff metadata
Marina: /nanopore/marina/t2t_v2/out-chm13-draft_v1-full-isoseq-chm13-only/analysis/consensus_v2/CHM13.combined.v2.bb

# build bed 
genePredToBed consensus_gene_set/CHM13.combined.v2.1.gp.gz  consensus_gene_set/CHM13.combined.v2.1.bed
pigz consensus_gene_set/CHM13.combined.v2.1.bed 

# 
# hard link CAT_V2 files
ln -f assemblyHub/CHM13/*.bam assemblyHub/CHM13/*.bai assemblyHub/CHM13/*.bb ../../../hub/t2t-chm13-v1.0/CAT_V2.1/
pigz consensus_gene_set/CHM13.combined.v2.1.bed 

# copy to globus
find CAT_V2.1 | cpio -pdl ../tmp/

* CAT_V3
2021-04-01 add in all paralogs from liftOff 

/nanopore/marina/t2t_v2/out-chm13-draft_v1-full-isoseq-chm13-only/analysis/consensus_v3

# make the bed
genePredToBed consensus_gene_set/CHM13.combined.v3.gp.gz  consensus_gene_set/CHM13.combined.v3.bed
pigz consensus_gene_set/CHM13.combined.v3.bed 

ln -f assemblyHub/CHM13/*.bam assemblyHub/CHM13/*.bai assemblyHub/CHM13/*.bb ../../../hub/t2t-chm13-v1.0/CAT_V3/
mkdir  ../../../hub/t2t-chm13-v1.0/CAT_V3/consensus_gene_set/
ln -f consensus_gene_set/* ../../../hub/t2t-chm13-v1.0/CAT_V3/consensus_gene_set/

* CAT_V4

mkdir -p assemblyHub/CHM13 assemblyHub/GRCh38/

ln -f ../CAT_V3/assemblyHub/CHM13/* assemblyHub/CHM13/
rm assemblyHub/CHM13/CHM13.combined.v3.bb 
ln -f consensus_gene_set/CHM13.combined.v4.bb assemblyHub/CHM13/
ln -f ../CAT/assemblyHub/GRCh38/* assemblyHub/GRCh38/
genePredToBed CHM13.combined.v4.gp.gz CHM13.combined.v4.bed

mkdir -p  ../../../hub/t2t-chm13-v1.0/CAT_V4/
ln -f assemblyHub/CHM13/*  ../../../hub/t2t-chm13-v1.0/CAT_V4/
find consensus_gene_set | cpio -pduml ../../../hub/t2t-chm13-v1.0/CAT_V4

cd CAT_V4/frameShifts/
../../../../gene-annotation/t2t-chm13-gene-analysis/bin/genePredFrameShifts --catMetaTsv=../consensus_gene_set/CHM13.combined.v4.gp_info.gz ../consensus_gene_set/CHM13.combined.v4.gp.gz CAT_V4.frameshifts.tsv CAT_V4.frameshifted.gp  CAT_V4.frameshifts.bed

* CpG islands:
cd cpgIslands

# this fails and only does hardmasked anyway
##doCpgIslands.pl -stop makeBed -buildDir /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/cpgIslands -maskedSeq /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/genome/t2t-chm13-v1.0.2bit  -chromSizes /hive/users/markd/nanopore/projs/t2t-chm13/build/t2t-chm13-v1.0/genome/t2t-chm13-v1.0.sizes t2t-chm13-v1.0 >&log&


twoBitToFa ../genome/t2t-chm13-v1.0.2bit stdout | /hive/data/staging/data/cpgIslandExt/cpglh /dev/stdin >cpgIslands.cpglh
./my-doMakeBed.csh 
ln t2t-chm13-v1.0.cpgIslandExt.bb ../../../hub/t2t-chm13-v1.0/


* readCoverage
** 2021-04-20 Arang Rhie
team-curation/tracks/v1/coverage/*.wig has the wig files. There are pri and markerandlength wig files for hifi and ont.
for f in *.wig ; do wigToBigWig -clip $f ../genome/t2t-chm13-v1.0.sizes $(basename $f .wig).bigWig ; done
ln -f *.bigWig ../../../hub/t2t-chm13-v1.0/readCoverage/

and set the track height max to 70 for hifi, 200 for ONT?

HiFi Pri: lightpink 255,182,193
HiFi Marker: lightcoral 240,128,128
ONT Pri: lightblue 173,216,230
ONT Marker: cornflowerblue 100,149,237

** 2021-05-05 Arang Rhie
I realized the marker assisted mapping had dropped the chrX! 
team-curation/tracks/v1/coverage/*.markersandlength.cov.wig

* low-coverage2
Arang Rhie
Can we retire the Low support tracks, and replace this to https://github.com/marbl/CHM13-issues/blob/main/v1.0/hifi.pri.issues.bed
The description.html is here: https://github.com/marbl/CHM13-issues/blob/main/description.html

** download
hifi.pri.clipped.bed
hifi.pri.issues.bed
ont.pri.len1k_idy85.clipped.bed
ont.pri.len1k_idy85.issues.bed
for f in *.gz ; do buildBigBed bed9 t2t-chm13-v1.0 $(basename $f .bed.gz).bigBed $f ; done
ln -f *.bigBed ../../../hub/t2t-chm13-v1.0/low_coverage2/

** replaced by issues
* proSeqUnstranded2
from Savannah

 /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_k100/PROseq/Not_stranded/
ln *.bigwig  ../../../hub/t2t-chm13-v1.0/proSeqUnstranded2/
* proSeqDefault
from Savannah
/team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_default/
ln -f *.bigwig  ../../../hub/t2t-chm13-v1.0/proSeqDefault

* proSeq3 update and merge

** Savannah: If so, Id like to combine PROseq_bt2_kmerFilt_notStranded and PROseq_bt2_kmerFilt_split-strands. 
Keeping all tracks,
but still retaining the blue/red colors for the split stranded ones and black for the notStranded (if you can do that). Can call it: PROseq_bt2-k100_kmerFilt
combine these:

PROseq_bt2_kmerFilt_split-strands
PROseq_bt2_kmerFilt_notStranded


** build track
# link in existing datafiles
ln -f ../proSeqSplitStrands2/* .
ln -f ../proSeqUnstranded2/* .

mkdir -p ../../../hub/t2t-chm13-v1.0/proSeq3
ln -f * ../../../hub/t2t-chm13-v1.0/proSeq3/
* proSeqDefault

** Savannah:
Id like to add more tracks to this superset PROseq_bt2-default_kmerFilt_notStranded. 

This will essentially be the same thing as the merged superset in (proSeq3), but
with different mapping parameters and will also include a mix of stranded
(red/blue colored) and not stranded (black).

Ive attached the track database and description.html for this. 
All files are here: /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_default/ 
Can rename to: PROseq_bt2-default_kmerFilt

add ones from /team-epigenetics/20200922-v1.0_chm13_PROseq/Bowtie2_default

** update existing directory and tracks
ln -f * ../../../dev/t2t-chm13-v1.0/proSeqDefault/

* rnaSeq3
** Savannah Finishing up the the last superset to be added, which will be RNAseq (none of which is stranded)

All files are here: /team-epigenetics/20200922-v1.0_chm13_PROseq/CHM13_RNAseq_analysis/Bowtie2_default/
Can call this:  RNAseq_bt2-default_kmerFilt
No special coloring here. All black tracks.

mkdir ../../../dev/t2t-chm13-v1.0/rnaSeq3
ln -f * ../../../dev/t2t-chm13-v1.0/rnaSeq3

* compositeRepeats
Savannah files set by slack and made available for download for the paper
http://t2t.gi.ucsc.edu/chm13/dev/t2t-chm13-v1.0/downloads/t2t-chm13-v1.0.composite-repeats-singletons-arrays.bed.gz

buildBigBed bed4 t2t-chm13-v1.0 compositeRepeats.bigBed  t2t-chm13-v1.0.composite-repeats-singletons-arrays.bed.gz 
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/compositeRepeats

* newSatellitesMonomersArrays
buildBigBed bed4 t2t-chm13-v1.0 newSatellitesMonomersArrays.bigBed t2t-chm13-v1.0.new-satellites-monomers-arrays.bed.gz 
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/newSatellitesMonomersArrays

Savannah files set by slack and made available for download for the paper
http://t2t.gi.ucsc.edu/chm13/dev/t2t-chm13-v1.0/downloads/t2t-chm13-v1.0.new-satellites-monomers-arrays.bed.gz

* to-do
** finish lastz



** Christopher Vollmers
@Mark Diekhans @Glennis Logsdon Here is what Mandalorion thinks the isoform sequences derived from your Iso-Seq reads are. The fasta files are straight consensus sequences generated from the subreads using PacBio consensus tools used slightly off-label. So these sequences should match CHM13 genome sequence. The psl file contains alignments of those consensus sequences to hg38 that are cleaned up, i.e. small indels are removed. One note to the output: Mandalorion will produce multiple isoforms with the same intron-chain if it thinks their ends are consistently different. Naming is Isoform_[running_number]_[number of Iso-Seq reads associated with the isoform].
chm13-isoseq--mandalorion/




** cactus liftover nets with synten
** liftover comparison



** hg38:
*** minmap chaining
*** cactus nets

**  RNA-seq data
Hi @Mark Diekhans quick question: did you (or anyone from your gene-team) ever map the ILMN RNA-seq data from @mydennis?


** winnowmap ONT
# currently for Karen, don't have track desc yet
/team-curation/alignments/20200921_v1.0/ont/
    ont.bam, ont.bam.bai

cd winnowmap_ont/
mkdir -p ../../../hub/t2t-chm13-v1.0/winnowmap_ont/
ln -f ont.bam*  ../../../hub/t2t-chm13-v1.0/winnowmap_ont/

** hg38 alignments





* Arang Hello @MarkD! I've pushed another update on the issues last night - it's now including the error k-mers found with Merqury and 3 additional errors found but forgotten :disappointed: . Please grep the v1.0_issues.bed and v1.1_issues.bed from here:
https://github.com/marbl/CHM13-issues
Description.html is also updated in this repo.

* alphaSatHOR tracks (top level at Lev's request) [for PUB]
Lev
The five annotation tracks for UCSC GB with the description files are on Globus at /team-satellites/AS_Annotation/release_v1.0/11Jul21/
Link: team-satellites/AS_Annotation/release_v1.0/11Jul21/
    ASat_HOR_track.bed
    ASat_SF_track.bed
    ASat_strand.bed
    ASat_StV10.bed
    Shared_ASat_HORs.bed

ASat_HOR_track.bed:track name="ASat HOR-track" description="Annotation of ASat HOR monomers (T2T CHM13 assembly release v1.0, HMMs 060721)" itemRgb="On" visibility="1"
ASat_SF_track.bed:track name="ASat SF-track" description="Annotation of ASat monomers into SF-classes (T2T CHM13 assembly release v1.0, HMMs June 29, 2021)" itemRgb="On" visibility="1"
ASat_strand.bed:track name="ASat strand" description="Annotation of ASat strand orientation (T2T CHM13 assembly release v1.0, HMMs 060721)" itemRgb="On" visibility="1"
ASat_StV10.bed:track name="StV10" description="Structural Variants 10" itemRgb="On" visibility="1"
Shared_ASat_HORs.bed:track name="Shared ASat HORs" description="Annotation of ASat double (13/21 and 14/22) and triple (1/5/19) HORs into chromosome-specific subtypes" itemRgb="On" visibility="1"


buildBigBed bed9 t2t-chm13-v1.0 ASat_HOR.bigBed ASat_HOR_track.bed
buildBigBed bed9 t2t-chm13-v1.0 ASat_SF.bigBed ASat_SF_track.bed
buildBigBed bed9 t2t-chm13-v1.0 ASat_strand.bigBed ASat_strand.bed
buildBigBed bed9 t2t-chm13-v1.0 ASat_StV10.bigBed ASat_StV10.bed
buildBigBed bed9 t2t-chm13-v1.0 Shared_ASat_HORs.bigBed Shared_ASat_HORs.bed


mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/alphaSatHOR 

* cytobands 2021-10-01
from @Nick Altemose via slack

chm13_v1.0.cytoBandIdeo_v1.0.txt
buildBigBed bed4 --as=${HOME}/kent/src/hg/lib/cytoBand.as   t2t-chm13-v1.0  cytoBandIdeo.bigBed chm13_v1.0.cytoBandIdeo_v1.0.txt
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/cytobands/

* issues
replaces low-coverage2

Arang Rhie
https://raw.githubusercontent.com/marbl/CHM13-issues/main/v1.0_issues.bed
https://raw.githubusercontent.com/marbl/CHM13-issues/main/description.html

buildBigBed bed9  t2t-chm13-v1.0 issues.bigBed v1.0_issues.bed 
mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/issues

move under validation supertrack

* merfinKmer (validation)
Giulio Formenti
team-curation/tracks/v1.0/chm13.draft_v1.0.hifi.bw
merfin_hifi_track.html
team-curation/tracks/v1.0/chm13.draft_v1.0.illumina.bw
merfin_illumina_track.html
mkln *.bw ../../../../hubs/dev/t2t-chm13-v1.0/merfinKmer


* todo 
Glennis Logsdon
Yeah, the initial Iso-Seq alignments were produced by Phillip Dishuck!

mkln *.bigBed ../../../../hubs/dev/t2t-chm13-v1.0/
