<h2>Description</h2>
<p>Merqury - k-mer multiplicity in 10XG reads</p>


<h2>Methods</h2>
<p>The first 23 bp of the 10X Genomics read pair 1 were trimmed to remove barcode bases,
and 21-mers were collected with Meryl (read set). The multiplicity was annotated along the first base of the k-mer on the assembly with meryl-lookup -dump.


Each beginning position (1-base) of the k-mer was annotated with the global copy number as a .wig file, and converted as bigWig format. As a result, the last 20 bases of an assembly are not included but follows the last 21 baseâ€™s copy number.
</p>


<h2>Display Conventions and Configuration</h2>
<!-- if there are special conventions, like colors, describe them here -->
<p>Maximum set to 100x as default. Adjust accordingly for repetitive regions.</p>


<h2>Data access</h2>
<!-- Optional, links for downloading the data if there is a separate source beside the
  browser  -->
<p>Data is available on globus: team-curation/merqury/20200602.read_multiplicity.bigWig with a corresponding .tdf</p>


<h2>Release history</h2>
<ol>
  <li> 2020 June 9 : Initial upload for 20200602 release
</ol>


<h2>Contact</h2>
<p>Contact <a href="mailto:rhiea@nih.gov">Arang Rhie &lt;rhiea@nih.gov&gt;</a>
</p>


<h2>Credits</h2>
<p></p>
