#+STARTUP: nologdone
#+SEQ_TODO: TODO ACTIVE | DONE

==== t2t-chm13-v1.0 public browser hub ====
* Info:
** URLs
https://hgwdev.gi.ucsc.edu/~markd/t2t/pub/hub.txt
https://t2t.gi.ucsc.edu/chm13/pub/hub.txt

- shared link
https://t2t.gi.ucsc.edu/chm13/hub/hub.txt

** most tracks copied from initial (now dev) build and updated.
** list of tracks to make public
https://docs.google.com/document/d/1QmOwTN_ay8_6S-Lo9Clcj7gLZTLGnBxNMBrNN7rScO0/edit
** build dir: projs/t2t-chm13/build/pub/t2t-chm13-v1.0
* tracks to add (as of 2021-05-25T12:54:57-0700)
** Known hets:
*** Curated SVs (.vcf) (XX)
*** Lightly filtered SNVs (.vcf) (XX)
*** Combined hets region (.bed) (XX)[a]
** Marker density (.wig or .bw) 
*** 21-mer (already up in the browser) (SK has description html)
*** 51-mer (already up in the browser)
*** 100-mer (team-curation/tracks/v1/chm13v1_unique_100mers_reformatted4overlaps.markerdesert.bed[b])
*** Min unique k-mer (team-epigenetics/052021_chm13_minUniqKmers) 
** Known issues:
*** Issues (.bed) (already in browser under low coverage)
*** Merqury - “asm only” (team-curation/tracks/v1/merqury/hybrid.missing.bed) (SK has html)
*** Corrections made from v1.0 to v1.1 (assemblies/release/v1.1/liftover/v1_patch) (SK has HTML)  
** Merfin k* tracks (.bw)
*** HiFi (.bw) team-curation/tracks/v1.0/chm13.draft_v1.0.hifi.bw (SK has HTML description)
*** Illumina PCR-free (.bw) team-curation/tracks/v1.0/chm13.draft_v1.0.illumina.bw 
** GRCh38 alignments
LINK TO DB BROWSER
*** Regions missing from GRCh38 (bed) (team-curation/tracks/v1/chm13_v1.0_uncoveredByGRCh38WinnowmapAlignments.bed) (SK has HTML)
*** Lifted over GRCh38 issues (team-variants/grch38_issues/hg38.parsedissues.bed) (SK has html)
*** GRCh38 synteny 1mb (already in browser)
*** Cactus alignments (already in browser)
*** LASTZ alignments (already in browser)
[don't break links in https://github.com/marbl/CHM13]
** ProSeq (already in browser, confirm no need to remove any)
** RNAseq (already in browser, confirm no need to remove any)
** CutnRun (losalt CENP-A only) (already in browser) 

** Methylation frequency (already in browser)
** Human population data
*** HGDP WSSD (already in browser)
*** T2TDP WSSD (already in browser)
*** 1000G variants
*** HPRC+ variants
*** dbSNP variants
*** ClinVar
*** GWAS SNPs[c]
* Other tracks:
** chromosome bands: Nick Altemose
need to clear with NCBI
** Sergey Koren  5 hours ago
@MarkD see above for the 1000G variant call track info
https://t2t-consortium.slack.com/archives/C014HC8NCLW/p1622058780104900?thread_ts=1621882203.094500&cid=C014HC8NCLW
** making top-level ISO-seq
* Track Html from tracks.tar.gz 
** chm13_liftedoverGRCh38_issues.html
** chm13_uncovered_byGRCh38.html
** DONE gc_ga_ct_at_track.html
** mapping.html
** merfin_hifi_track.html
** merfin_illumina_track.html
** merqury_asmonly.html
** unique_21_51_mer.html
** v1.0_to_v1.1.html
* Track build log
`** rmsk
from Savannah /team-TE/Repeatmasker_polished/chm13-v1_Repeatmasker_polish_031921_track_v2.bed
ln -f ../../../dev/t2t-chm13-v1.0/rmskV2/chm13v1_polish-033121_track.bed.gz .
ln -f ../../../dev/t2t-chm13-v1.0/rmskV2/rmskV2.bed.gz rmsk.bed.gz 
ln -f ../../../dev/t2t-chm13-v1.0/rmskV2/rmskV2.bigBed rmsk.bigBed 
ln -f rmsk.bigBed ../../../../pub/t2t-chm13-v1.0/rmsk/
** trf
from Mitchell Robert Vollger
/team-segdups/Assembly_analysis/Masked/chm13.draft_v1.0_plus38Y_trf.bed
ln -f ../../../dev/t2t-chm13-v1.0/trf/trf.bigBed .
ln -f trf.bigBed ../../../../pub/t2t-chm13-v1.0/trf/
** genome
note: was never updated to latest repeat masker; but leave as-is to match release
ln -f ../../../dev/t2t-chm13-v1.0/genome/* .

mkln t2t-chm13-v1.0.*  ../../../../pub/t2t-chm13-v1.0/genome/
** gc5Base
ln -f ../../../dev/t2t-chm13-v1.0/gc5Base/gc5Base.* .
mkln gc5Base.bigWig  ../../../../pub/t2t-chm13-v1.0/gc5Base
** gap
ln -f ../../../dev/t2t-chm13-v1.0/gap/* .
mkln gap.bigBed  ../../../../pub/t2t-chm13-v1.0/gap

** cpgIslands
ln -f ../../../dev/t2t-chm13-v1.0/cpgIslands/t2t-chm13-v1.0.cpgIslandExt.bb .
** sedefSegDups
ln -f ../../../dev/t2t-chm13-v1.0/sedefSegDups/chm13.draft_v1.0_plus38Y.SDs.* .
mkln * ../../../../pub/t2t-chm13-v1.0/sedefSegDups
** cenSat
ln -f ../../../dev/t2t-chm13-v1.0/cenSatV2/* .
mkln *.bigBed ../../../../pub/t2t-chm13-v1.0/cenSat
** CAT
(cd   ../../../dev/t2t-chm13-v1.0/CAT_V4/ && find . | cpio -pduml ../../../pub/t2t-chm13-v1.0/CAT/)

mkln assemblyHub/CHM13/* ../../../../pub/t2t-chm13-v1.0/CAT/
mkln consensus_gene_set/*  ../../../../pub/t2t-chm13-v1.0/CAT/consensus_gene_set/
** microSatRepeats
team-curation/tracks/v1/pattern/*.wig
v1.AT.128.wig  v1.GA.128.wig  v1.GC.128.wig  v1.TC.128.wig
pigz *.wig
for f in *.gz ; do (wigToBigWig $f ../genome/t2t-chm13-v1.0.sizes $(basename $f .wig.gz).bigWig &) ; done
mkln *.bigWig  ../../../../pub/t2t-chm13-v1.0/microSatRepeats/

** ACTIVE winnowmapAlign  Winnowmap alignments (v2.01): primary alignments
*** HiFi (team-curation/alignments/20200921_v1.0/hifi/*.wig, med when multiple available) (SK has description)

**** 1st and 2nd allele frequency (.bw)
https://eichlerlab.gs.washington.edu/help/glogsdon/Shared/210413_allchrs.mostCommonBase.bw
https://eichlerlab.gs.washington.edu/help/glogsdon/Shared/210413_allchrs.2ndmostCommonBase.bw
*** ONT (team-curation/alignments/20200921_v1.0/ont/*[pn].wig, med when multiple available) (SK has description)
**** Coverage (per-strand)
**** Med. Idy (per-strand)
**** Med. read length (per-strand)
**** Med. MQ (per-strand)
**** Med. reads per strand (%)


**  winnowmapAlign
*** HiFi
Arang Rhie
team-curation/alignments/20200921_v1.0/hifi/*.wig
- Coverage (unstranded) hifi.pri.cov.wig
- Med. Idy (unstranded) hifi.pri.idy.med.wig
- Med. read length (unstranded) hifi.pri.len.med.wig
- Med. MQ (unstranded) hifi.pri.mq.med.wig
- 1st and 2nd allele frequency
https://eichlerlab.gs.washington.edu/help/glogsdon/Shared/210413_allchrs.mostCommonBase.bw
https://eichlerlab.gs.washington.edu/help/glogsdon/Shared/210413_allchrs.2ndmostCommonBase.bw

* MAKE SURE THESE WORK
https://github.com/marbl/CHM13
hg38 alignments, cat gff
