* general stuff
t2t-chm13.20200727  v1.0
** URLs
https://hgwdev.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
https://courtyard.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
https://t2t.gi.ucsc.edu/chm13/hub/hub.txt

* PATH setup
export PATH=${HOME}/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${PATH}
* genome assembly
# 2020-07-28
/assemblies/drafts/t2t-chm13.20200727.fasta  # include chrMT
bgzip --threads=64 t2t-chm13.20200727.fasta
makeMd5s t2t-chm13.20200727.fasta.gz 
ln t2t-chm13.20200727.fasta.gz t2tChm13_20200727.unmasked.fa.gz
faToTwoBit t2tChm13_20200727.unmasked.fa.gz t2tChm13_20200727.unmasked.2bit


* rmsk
2020-07-28 from Mitchell Robert Vollger
team-segdups/Masker/t2t-chm13.20200727_repeatmasker.out
cd rmsk
bgzip --threads=65 t2t-chm13.20200727_repeatmasker.out 
makeMd5s t2t-chm13.20200727_repeatmasker.out.gz 
(../../../T2T-CHM13-hub/bin/RM2Bed.py t2t-chm13.20200727_repeatmasker.out.gz - | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin stdout | gzip -c >rmsk.bed.gz) >&log&
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskBed.as --twoBit=../genome/t2tChm13_20200727.unmasked.2bit bed9+ t2tChm13_20200727 rmsk.bigBed rmsk.bed.gz
ln -f rmsk.bigBed ../../../hub/t2tChm13_20200727/

* trf
2020-07-28 from Mitchell Robert Vollger
team-segdups/Masker/t2t-t2t-chm13.20200727_repeatmasker.out
gzip t2t-chm13.20200727_trf.bed
makeMd5s t2t-chm13.20200727_trf.bed.gz 
buildBigBed --as=../../../T2T-CHM13-hub/etc/simpleRepeat2.as --twoBit=../genome/t2tChm13_20200727.unmasked.2bit bed3+ t2tChm13_20200727 trf.bigBed t2t-chm13.20200727_trf.bed.gz 
ln -f trf.bigBed  ../../../hub/t2tChm13_20200727/

* mask and finish genome
2020-07-28
cd genome
twoBitMask -type=.bed t2tChm13_20200727.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/t2t-chm13.20200727_trf.bed.gz ) t2tChm13_20200727.2bit
twoBitInfo t2tChm13_20200727.2bit t2tChm13_20200727.sizes
twoBitToFa t2tChm13_20200727.2bit t2tChm13_20200727.fa
nice bgzip  --threads=64 t2tChm13_20200727.fa
samtools faidx t2tChm13_20200727.fa.gz 
blat t2tChm13_20200727.2bit -makeOoc=t2tChm13_20200727.11.ooc /dev/null /dev/null
ln -f t2tChm13_20200727.2bit  ../../../hub/t2tChm13_20200727/
ln -f t2tChm13_20200727.fa.gz*  ../../../hub/t2tChm13_20200727/


* sedefSegDups/sedefSegDupsLowId
2020-07-29 from Mitchell Robert Vollger
have made the SEDEF segdup tracks  for the new asm if you want to add them to the browser.
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.browser.bed
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.browser.html
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.lowid.browser.bed


bgzip --threads=64 *.bed 
makeMd5s *.bed.gz
buildBigBed bed9 --as=../../../T2T-CHM13-hub/etc/sedefSegDups.as t2tChm13_20200727 sedefSegDups.bigBed SDs.browser.bed.gz
buildBigBed bed9 --as=../../../T2T-CHM13-hub/etc/sedefSegDups.as t2tChm13_20200727 sedefSegDupsLow.bigBed SDs.lowid.browser.bed.gz
ln -f *.bigBed   ../../../hub/t2tChm13_20200727/sedefSegDups/

2020-07-31 rebuilt due to column name problems


* as_annotation (as_hor_annotation, as_sf_annotation)
2020-07-31 from Lev Uralsky
cd as_annotation
/team-satellites/AS_Annotation/draft_20200727/
    AS_HOR_Annot.html -> as_hor_annotation.html
    AS_SF_Annot.html  -> as_sf_annotation.html
    t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm.h.bed.gz
    t2t-chm13.20200727-vs-AS-SFs-hmmer3.0-30072020.hmm.h.bed.gz

makeMd5s *.bed.gz

buildBigBed bed9 t2tChm13_20200727 as_hor_annotation.bigBed t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm.v2.h.bed.gz
buildBigBed bed9 t2tChm13_20200727 as_sf_annotation.bigBed t2t-chm13.20200727-vs-AS-SFs-hmmer3.0-30072020.hmm.h.bed.gz 

ln -f as_*annotation.bigBed  ../../../hub/t2tChm13_20200727/

# updated 2020-08-02 due to error found in data


* gap track 2020-08-02
../../../T2T-CHM13-hub/bin/faToGapBed ../genome/t2tChm13_20200727.fa.gz gap.bed
buildBigBed bed3 t2tChm13_20200727 gap.bigBed gap.bed
ln -f gap.bigBed  ../../../hub/t2tChm13_20200727/

* hor_decompositions
update from Tatiana Dvorkina 2020-08-10
/team-satellites/HorAnnotation/20200727/hordecomposition_cen*_v1.0.bed
gzip -9 *.bed
makeMd5s *.bed.gz
buildBigBed bed9 t2tChm13_20200727 hor_decompositions.bigBed *.bed.gz
ln -f hor_decompositions.bigBed   ../../../hub/t2tChm13_20200727/
updated 2020-08-11 due to error

* hor_decompositions_native
update from Tatiana Dvorkina 2020-08-19
team-satellites/HorAnnotation/20200727_naive/
gzip -9 *.bed
makeMd5s *.bed.gz
buildBigBed bed9 t2tChm13_20200727 hor_decompositions_native.bigBed *.bed.gz
ln -f hor_decompositions_native.bigBed   ../../../hub/t2tChm13_20200727/

* winnowmap_ont_all
Chirag Jain
team-curation/alignments/20200727/ONT/rel5_all_include_uw/output.bam
team-curation/alignments/20200727/ONT/rel5_all_include_uw/output.filtered.sniffles.vcf

bgzip --threads=64 output.filtered.sniffles.vcf
makeMd5s output.*

ln -f output.bam winnowmap_ont_all.bam
ln -f output.bam.bai winnowmap_ont_all.bam.bai

vcf-sort output.filtered.sniffles.vcf.gz |bgzip --threads=64 -c > winnowmap_ont_all_variants.vcf.gz
tabix -p vcf winnowmap_ont_all_variants.vcf.gz 

ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_ont/


* winnowmap_ont_ul
Chirag Jain
team-curation/alignments/20200727/ONT/rel5_over_100k/output.bam
makeMd5s output.* 
ln -f output.bam winnowmap_ont_ul.bam
ln -f output.bam.bai winnowmap_ont_ul.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_ont

* winnowmap_clr
Chirag Jain
team-curation/alignments/20200727/clr/output.bam
makeMd5s output.* &
ln -f output.bam winnowmap_pb_clr.bam
ln -f output.bam.bai winnowmap_pb_clr.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_pb

* winnowmap_hifi
Chirag Jain
team-curation/alignments/20200727/hifi/output.bam
makeMd5s output.* &
ln -f output.bam winnowmap_pb_hifi.bam
ln -f output.bam.bai winnowmap_pb_hifi.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_pb

* illumina_pcrfree
Arkarachai Fungtammasan
team-curation/alignments/20200727/PCRfree/CHM13_markdup.bam
team-curation/alignments/20200727/PCRfree/deepvariant.vcf.gz
makeMd5s CHM13_markdup.bam* deepvariant.vcf.gz*&
ln -f CHM13_markdup.bam illumina_pcrfree.bam
ln -f CHM13_markdup.bam.bai illumina_pcrfree.bam.bai
ln -f deepvariant.vcf.gz  illumina_pcrfree_variants.vcf.gz
ln -f deepvariant.vcf.gz.tbi  illumina_pcrfree_variants.vcf.gz.tbi
ln -f illumina_pcrfree* ../../../hub/t2tChm13_20200727/illumina_pcrfree/

* bionano_cover
Arang Rhie
team-curation/alignments/20200727/bionano/mol/BN.cov.wig
bgzip --threads=64 BN.cov.wig 
makeMd5s BN.cov.wig.gz &
wigToBigWig -clip BN.cov.wig.gz ../genome/t2tChm13_20200727.sizes bionano_cover.bigWig
ln -f bionano_cover.bigWig ../../../hub/t2tChm13_20200727/


* kmer_metric
Arang Rhie
team-curation/kstar/t2t-chm13.20200727.kstar.bw
makeMd5s t2t-chm13.20200727.kstar.bw &
ln -f t2t-chm13.20200727.kstar.bw kmer_metric.bigWig
ln -f kmer_metric.bigWig ../../../hub/t2tChm13_20200727/

* hg38Lastz
cd hg38Lastz
ln -s ../../../T2T-CHM13-hub/build/t2tChm13_20200727/hg38Lastz.DEF DEF
runHg38Lastz t2tChm13_20200727 >&log&
# OPPS built chains backwards:
chainSwap axtChain/hg38.t2tChm13_20200727.all.chain.gz stdout | bgzip --threads=64 -c >axtChain/t2tChm13_20200727.hg38.all.chain.gz 

# create all bigChain
cd bigAll.tmp
hgLoadChain -noBin -test no bigChain ../axtChain/t2tChm13_20200727.hg38.all.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp.bigChain
bbedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp.bigChain ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp.bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp.bigLink  ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all.bigLink
cd ..
ln -f t2tChm13_20200727.hg38.all.big* ../../../hub/t2tChm13_20200727/grch38/


# files are 
hg38.t2tChm13_20200727.all.chain.gz
hg38.t2tChm13_20200727.net.axt.gz
hg38.t2tChm13_20200727.over.chain.gz


* cenRegions
Karen Miga
team-satellite/t2t-chm13.20200727.cenRegions.bed

bgzip --threads=64 t2t-chm13.20200727.cenRegions.bed
makeMd5s t2t-chm13.20200727.cenRegions.bed.gz 
buildBigBed bed8 t2tChm13_20200727 cenRegions.bigBed t2t-chm13.20200727.cenRegions.bed.gz
ln -f cenRegions.bigBed ../../../hub/t2tChm13_20200727

* cenSat_annotation
Karen Miga
team-satellite/t2t-chm13.20200727.cenSat_annotation.bed
bgzip --threads=64 t2t-chm13.20200727.cenSat_annotation.bed 
buildBigBed bed9 t2tChm13_20200727 cenSat_annotation.bigBed t2t-chm13.20200727.cenSat_annotation.bed.gz 
ln -f cenSat_annotation.bigBed ../../../hub/t2tChm13_20200727



* GC percent 2020-09-03
cd gc5Base
hgGcPercent -noLoad -wigOut -doGaps -win=5 -file=stdout -verbose=0 NODB ../genome/t2tChm13_20200727.2bit  | gzip -c > gc5Base.wigVarStep.gz
wigToBigWig gc5Base.wigVarStep.gz  ../genome/t2tChm13_20200727.sizes gc5Base.bigWig
ln gc5Base.bigWig  ../../../hub/t2tChm13_20200727/

* TransMap all chains (refseq and gencode) 
export PATH=${HOME}/compbio/projs/transMap/transMap-dev/bin:${PATH}
cd transMap/tmp


srcDbLoadEnsemblAligns --gencodeVersion=V35 hg38 ensembl data/src/hg38/hg38.ensembl.src.db
srcDbLoadEnsemblMetadata --gencodeVersion=V35 hg38 ensembl data/src/hg38/hg38.ensembl.src.db
srcDbLoadEnsemblSeqs hg38 ensembl data/src/hg38/hg38.ensembl.src.db  

srcDbLoadGenbankAligns hg38 refseq data/src/hg38/hg38.refseq.src.db
srcDbLoadGenbankMetadata hg38 refseq data/src/hg38/hg38.refseq.src.db
srcDbLoadGenbankSeqs hg38 refseq data/src/hg38/hg38.refseq.src.db  

# manually create
sqlite3 data/genome.db < ../../../../T2T-CHM13-hub/build/t2tChm13_20200727/transMapGenomeDb.sql 

# build with all chains
#  CLUSTER DOWN DUE TO FIRE, paraHost=hgwdev
ln -s  ../../../../T2T-CHM13-hub/build/t2tChm13_20200727/transMapBuildConf.py .
mappingChainBuild data/genome.db hg38 t2tChm13_20200727 all data/chains/hg38/t2tChm13_20200727/hg38.t2tChm13_20200727.all.chain
transMapBatch ./transMapBuildConf.py hg38 t2tChm13_20200727 ensembl all >&ens.log &
transMapBatch ./transMapBuildConf.py hg38 t2tChm13_20200727 refseq all >&ref.log &

transMapBuildBigPsl ./transMapBuildConf.py t2tChm13_20200727 ensembl hg38 >&ens.log &
transMapBuildBigPsl ./transMapBuildConf.py t2tChm13_20200727 refseq hg38 >&ref.log &

cd ..
ln tmp/results/mapped/t2tChm13_20200727/t2tChm13_20200727.ensembl.transMapV1.bigPsl t2tChm13_20200727.ensembl.transMap.all.bigPsl 
ln tmp/results/mapped/t2tChm13_20200727/t2tChm13_20200727.refseq.transMapV1.bigPsl  t2tChm13_20200727.refseq.transMap.all.bigPsl 
mkdir ../../../hub/t2tChm13_20200727/transMap
ln t2tChm13_20200727.*.all.bigPsl ../../../hub/t2tChm13_20200727/transMap/

## maybe add trixIdx to make case-insensitive: ixIxx input.txt myTrix.ix myTrix.ixx

* shortRnaSeqIntrons:
cd shortRnaSeq
see t2t-chm13-gene-analysis/doc/t2tChm13_20200727.org for details
on how it was built.

buildBigBed bed12 t2tChm13_20200727 shortRnaSeqIntrons.bigBed ../../../gene-annotation/build/t2tChm13_20200727/shortRnaSeq/raw-introns-combined/shortRnaSeq-2020-08-28.juncs.bed.gz 
ln shortRnaSeqIntrons.bigBed ../../../hub/t2tChm13_20200727/

* refSeqBlat
cd refSeqBlat/data
# files from 2020-05-29
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_knownrefseq_alignments.bam
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.gbff.gz
faSize -detailed GRCh38_latest_rna.fna.gz | tawk '/^N/{print $1}' >refSeq-curated.acc
faSomeRecords GRCh38_latest_rna.fna.gz refSeq-curated.acc stdout | bgzip -@ 64 -c >refSeq-curated.fa.gz

# metadata for later merge (2m16s)
gbffToTsv data/GRCh38_latest_rna.gbff.gz data/refSeq-curated.meta.tsv.gz >&log&

cd ..
blatRnaMkJobs ../genome/t2tChm13_20200727.2bit data/refSeq-curated.fa.gz tmp/
para make -batch=tmp/b1 tmp/para.jobs >&log&

# filter 
faPolyASizes data/GRCh38_latest_rna.fna.gz stdout | gzip -c >data/GRCh38_latest_rna.polya.gz

find tmp/psls -type f | xargs cat | sort -k 15,15 | pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -localNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=refSeq-curated.stats stdin stdout | sort -k 14,14 -k 16,16n |gzip -9c >refSeq-curated.psl.gz

# build bigPsl with extra columns
zcat data/refSeq-curated.meta.tsv.gz | tawk '$6!=""{print $1,$6}' > tmp/rs.cds
pslToBigPsl -cds=tmp/rs.cds -fa=data/GRCh38_latest_rna.fna.gz refSeq-curated.psl.gz stdout | sort -k1,1 -k2,2n > tmp/rs.bigin1
bigBlatPslAddColumns data/refSeq-curated.meta.tsv.gz tmp/rs.bigin1 tmp/rs.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/rs.bigin2 ../genome/t2tChm13_20200727.sizes refSeq-curated.bigBlatPsl


* synteny  2020-09-04
cd synteny/tmp
halStats --bedSequences t2tChm13_20200727../../../data/primates_eichler.hal > chm13.bed

create syn.tmpl:
#LOOP
../../../../T2T-CHM13-hub/bin/halSyntenyRun ../../CAT/assemblyHub/t2tChm13.hal t2tChm13_20200727 $(path1) GRCh38 25000 25000 {check out exists out/25kb.$(path1).psl}
../../../../T2T-CHM13-hub/bin/halSyntenyRun ../../CAT/assemblyHub/t2tChm13.hal t2tChm13_20200727 $(path1) GRCh38 1000000 50000 {check out exists out/1mb.$(path1).psl}
#ENDLOOP

gensub2 <(cut -f 1 chm13.bed) <(echo "") syn.tmpl syn.jobs

 # run with -ram=16g
 para create -batch=b1 syn.jobs -ram=16g -maxQueue=100000000

# halSyntenty produced no output for chrMT, which is identical

# combine data for tracks
cat  out/25kb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >../synteny.25kb.psl.gz
cat  out/1mb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >../synteny.1mb.psl.gz

pslToBigPsl ../synteny.25kb.psl.gz 25kb.bigin
pslToBigPsl ../synteny.1mb.psl.gz 1mb.bigin
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 25kb.bigin ../../genome/t2tChm13_20200727.sizes ../synteny.25kb.bigPsl
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 1mb.bigin ../../genome/t2tChm13_20200727.sizes ../synteny.1mb.bigPsl

ln  *.bigPsl ../../../hub/t2tChm13_20200727/synteny/

** CAT annotations 2020-09-03
# run by Marina in AWS and downloaded it to: 
   courtyard.gi.ucsc.edu:/nanopore/marina/out-chm13-t2t/assemblyHub


cd CAT
rsync -a courtyard.gi.ucsc.edu:/nanopore/marina/out-chm13-t2t/assemblyHub . >&log&

# need to create a GRCh38/CAT and add to GRCh38 to assembly hub
ln assemblyHub/GRCh38/annotation.bb assemblyHub/GRCh38/GRCh38.2bit  ../../../hub/GRCh38/

ln assemblyHub/CHM13/*.bam ../../../hub/t2tChm13_20200727/CAT/
ln assemblyHub/CHM13/*.bam.bai ../../../hub/t2tChm13_20200727/CAT/
ln assemblyHub/CHM13/*.bb   ../../../hub/t2tChm13_20200727/CAT/

# rename HAL
export PATH=${HOME}/compbio/compartiveGenomics/projs/hal/src/hal/bin:${PATH}
CHM13	t2tChm13_20200727
halRenameGenomes assemblyHub/t2tChm13.hal rename.tab
halStats --genomes assemblyHub/t2tChm13.hal
ln assemblyHub/t2tChm13.hal   ../../../hub/t2tChm13_20200727/CAT/


* todo
** RMMaskBed to David, etc
** as_sf_annotation.html as_hor_annotation.html reference a google doc
** chains for hg38->t2t, announce to variant group.


** introns /team-genes/20200727/shortRnaSeq/raw-introns
** GENCODE and RefSeq BLAT alignments
** mydennis  CN mapps
We generated CN maps of the 20200727 release using kmers and CHM13 genome Illumina data (quickmer2 pipeline). Not sure if this would be of interest to add as a track
track name=CHM13_CN itemRgb="On" bigDataUrl=https://bioshare.bioinformatics.ucdavis.edu/bioshare/download/rlc692m7tk5cibb/t2t/t2t_chm13_20200727/quickmer2/SRR3986881.depth.bed.CN.bb type=bigBed
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7073954/
** proseq  (@Savannah Klein @Rachel O'Neill) 
**  RNA-seq PE ILMN mydennis
** all isoset 
** check default assembly in CHM13
*** report hide bug

