* general stuff
t2t-chm13.20200727
** URLs
https://hgwdev.gi.ucsc.edu/~markd/t2t/T2T-CHM13/hub.txt
Vhttps://t2t.gi.ucsc.edu/chm13/hub/hub.txt

* PATH setup
export PATH=${HOME}/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${PATH}
* genome assembly
# 2020-07-28
/assemblies/drafts/t2t-chm13.20200727.fasta  # include chrMT
bgzip --threads=64 t2t-chm13.20200727.fasta
makeMd5s t2t-chm13.20200727.fasta.gz 
ln t2t-chm13.20200727.fasta.gz t2tChm13_20200727.unmasked.fa.gz
faToTwoBit t2tChm13_20200727.unmasked.fa.gz t2tChm13_20200727.unmasked.2bit


* rmsk
2020-07-28 from Mitchell Robert Vollger
team-segdups/Masker/t2t-chm13.20200727_repeatmasker.out
cd rmsk
bgzip --threads=65 t2t-chm13.20200727_repeatmasker.out 
makeMd5s t2t-chm13.20200727_repeatmasker.out.gz 
(../../../T2T-CHM13-hub/bin/RM2Bed.py t2t-chm13.20200727_repeatmasker.out.gz - | ../../../T2T-CHM13-hub/bin/editRepeatMaskerBed | bedSort stdin stdout | gzip -c >rmsk.bed.gz) >&log&
buildBigBed  --as=../../../T2T-CHM13-hub/etc/rmskBed.as --twoBit=../genome/t2tChm13_20200727.unmasked.2bit bed9+ t2tChm13_20200727 rmsk.bigBed rmsk.bed.gz
ln -f rmsk.bigBed ../../../hub/t2tChm13_20200727/

* trf
2020-07-28 from Mitchell Robert Vollger
team-segdups/Masker/t2t-t2t-chm13.20200727_repeatmasker.out
gzip t2t-chm13.20200727_trf.bed
makeMd5s t2t-chm13.20200727_trf.bed.gz 
buildBigBed --as=../../../T2T-CHM13-hub/etc/simpleRepeat2.as --twoBit=../genome/t2tChm13_20200727.unmasked.2bit bed3+ t2tChm13_20200727 trf.bigBed t2t-chm13.20200727_trf.bed.gz 
ln -f trf.bigBed  ../../../hub/t2tChm13_20200727/

* mask and finish genome
2020-07-28
cd genome
twoBitMask -type=.bed t2tChm13_20200727.unmasked.2bit <(zcat ../rmsk/rmsk.bed.gz ../trf/t2t-chm13.20200727_trf.bed.gz ) t2tChm13_20200727.2bit
twoBitInfo t2tChm13_20200727.2bit t2tChm13_20200727.sizes
twoBitToFa t2tChm13_20200727.2bit t2tChm13_20200727.fa
nice bgzip  --threads=64 t2tChm13_20200727.fa
samtools faidx t2tChm13_20200727.fa.gz 
blat t2tChm13_20200727.2bit -makeOoc=t2tChm13_20200727.11.ooc /dev/null /dev/null
ln -f t2tChm13_20200727.2bit  ../../../hub/t2tChm13_20200727/
ln -f t2tChm13_20200727.fa.gz*  ../../../hub/t2tChm13_20200727/


* sedefSegDups/sedefSegDupsLowId
2020-07-29 from Mitchell Robert Vollger
have made the SEDEF segdup tracks  for the new asm if you want to add them to the browser.
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.browser.bed
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.browser.html
   team-segdups/Masker/t2t-chm13.20200727_sedef_out/SDs.lowid.browser.bed


gzip *.bed
makeMd5s *.bed.gz
buildBigBed bed9 --as=../../../T2T-CHM13-hub/etc/sedefSegDups.as t2tChm13_20200727 sedefSegDups.bigBed t2t-chm13.20200727.SDs.bed.gz
buildBigBed bed9 --as=../../../T2T-CHM13-hub/etc/sedefSegDups.as t2tChm13_20200727 sedefSegDupsLow.bigBed t2t-chm13.20200727.SDs.lowid.bed.gz
ln -f *.bigBed   ../../../hub/t2tChm13_20200727/sedefSegDups/

2020-07-31 rebuilt due to column name problems
2020-09-08 updated due to being rerun by UQ
 needed to add:    int sat_bases; "sat_bases" to schema

* as_annotation (as_hor_annotation, as_sf_annotation)
2020-07-31 from Lev Uralsky
cd as_annotation
/team-satellites/AS_Annotation/draft_20200727/
    AS_HOR_Annot.html -> as_hor_annotation.html
    AS_SF_Annot.html  -> as_sf_annotation.html
    t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm.h.bed.gz
    t2t-chm13.20200727-vs-AS-SFs-hmmer3.0-30072020.hmm.h.bed.gz

makeMd5s *.bed.gz

buildBigBed bed9 t2tChm13_20200727 as_hor_annotation.bigBed t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm.v2.h.bed.gz
buildBigBed bed9 t2tChm13_20200727 as_sf_annotation.bigBed t2t-chm13.20200727-vs-AS-SFs-hmmer3.0-30072020.hmm.h.bed.gz 

ln -f as_*annotation.bigBed  ../../../hub/t2tChm13_20200727/

# updated 2020-08-02 due to error found in data


* as_strand
2020-09-18 from Lev via google
t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm-tbl.strand.bed

buildBigBed bed9 t2tChm13_20200727 as_strand.bigBed t2t-chm13.20200727-vs-AS-HORs-hmmer3.0-30072020.hmm-tbl.strand.bed.gz
ln -f as_strand.bigBed  ../../../hub/t2tChm13_20200727/


* gap track 2020-08-02
../../../T2T-CHM13-hub/bin/faToGapBed ../genome/t2tChm13_20200727.fa.gz gap.bed
buildBigBed bed3 t2tChm13_20200727 gap.bigBed gap.bed
ln -f gap.bigBed  ../../../hub/t2tChm13_20200727/

* hor_decompositions
update from Tatiana Dvorkina 2020-08-10
/team-satellites/HorAnnotation/20200727/hordecomposition_cen*_v1.0.bed
gzip -9 *.bed
makeMd5s *.bed.gz
buildBigBed bed9 t2tChm13_20200727 hor_decompositions.bigBed *.bed.gz
ln -f hor_decompositions.bigBed   ../../../hub/t2tChm13_20200727/
updated 2020-08-11 due to error

* hor_decompositions_native
update from Tatiana Dvorkina 2020-08-19
team-satellites/HorAnnotation/20200727_naive/
gzip -9 *.bed
makeMd5s *.bed.gz
buildBigBed bed9 t2tChm13_20200727 hor_decompositions_native.bigBed *.bed.gz
ln -f hor_decompositions_native.bigBed   ../../../hub/t2tChm13_20200727/

* winnowmap_ont_all
Chirag Jain
team-curation/alignments/20200727/ONT/rel5_all_include_uw/output.bam
team-curation/alignments/20200727/ONT/rel5_all_include_uw/output.filtered.sniffles.vcf

bgzip --threads=64 output.filtered.sniffles.vcf
makeMd5s output.*

ln -f output.bam winnowmap_ont_all.bam
ln -f output.bam.bai winnowmap_ont_all.bam.bai

vcf-sort output.filtered.sniffles.vcf.gz |bgzip --threads=64 -c > winnowmap_ont_all_variants.vcf.gz
tabix -p vcf winnowmap_ont_all_variants.vcf.gz 

ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_ont/


* winnowmap_ont_ul
Chirag Jain
team-curation/alignments/20200727/ONT/rel5_over_100k/output.bam
makeMd5s output.* 
ln -f output.bam winnowmap_ont_ul.bam
ln -f output.bam.bai winnowmap_ont_ul.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_ont

* winnowmap_clr
Chirag Jain
team-curation/alignments/20200727/clr/output.bam
makeMd5s output.* &
ln -f output.bam winnowmap_pb_clr.bam
ln -f output.bam.bai winnowmap_pb_clr.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_pb

* winnowmap_hifi
Chirag Jain
team-curation/alignments/20200727/hifi/output.bam
makeMd5s output.* &
ln -f output.bam winnowmap_pb_hifi.bam
ln -f output.bam.bai winnowmap_pb_hifi.bam.bai
ln -f winnowmap_* ../../../hub/t2tChm13_20200727/winnowmap_pb

* illumina_pcrfree
Arkarachai Fungtammasan
team-curation/alignments/20200727/PCRfree/CHM13_markdup.bam
team-curation/alignments/20200727/PCRfree/deepvariant.vcf.gz
makeMd5s CHM13_markdup.bam* deepvariant.vcf.gz*&
ln -f CHM13_markdup.bam illumina_pcrfree.bam
ln -f CHM13_markdup.bam.bai illumina_pcrfree.bam.bai
ln -f deepvariant.vcf.gz  illumina_pcrfree_variants.vcf.gz
ln -f deepvariant.vcf.gz.tbi  illumina_pcrfree_variants.vcf.gz.tbi
ln -f illumina_pcrfree* ../../../hub/t2tChm13_20200727/illumina_pcrfree/

* bionano_cover
Arang Rhie
team-curation/alignments/20200727/bionano/mol/BN.cov.wig
bgzip --threads=64 BN.cov.wig 
makeMd5s BN.cov.wig.gz &
wigToBigWig -clip BN.cov.wig.gz ../genome/t2tChm13_20200727.sizes bionano_cover.bigWig
ln -f bionano_cover.bigWig ../../../hub/t2tChm13_20200727/


* kmer_metric
Arang Rhie
team-curation/kstar/t2t-chm13.20200727.kstar.bw
makeMd5s t2t-chm13.20200727.kstar.bw &
ln -f t2t-chm13.20200727.kstar.bw kmer_metric.bigWig
ln -f kmer_metric.bigWig ../../../hub/t2tChm13_20200727/

* hg38Lastz

** lastz run
# WARNING have query and target reverse, so had to build swap
# ALSO seems to run at priority 15
cd hg38Lastz
ln -s ../../../T2T-CHM13-hub/build/t2tChm13_20200727/hg38Lastz.DEF DEF
runHg38Lastz t2tChm13_20200727 >&log&
# OPPS built chains backwards:
chainSwap axtChain/hg38.t2tChm13_20200727.all.chain.gz stdout | bgzip --threads=64 -c >axtChain/t2tChm13_20200727.hg38.all.chain.gz 

# create all bigChain
cd bigAll.tmp
hgLoadChain -noBin -test no bigChain ../axtChain/t2tChm13_20200727.hg38.all.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp.bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp.bigChain ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp.bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp.bigLink  ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all.bigLink
cd ..
ln -f t2tChm13_20200727.hg38.all.big* ../../../hub/t2tChm13_20200727/grch38/


# files are 
hg38.t2tChm13_20200727.all.chain.gz
hg38.t2tChm13_20200727.over.chain.gz

chainSwap to
t2tChm13_20200727.hg38.all.chain.gz
t2tChm13_20200727.hg38.over.chain.gz

cd tmp
hgLoadChain -noBin -test no bigChain ../axtChain/t2tChm13_20200727.hg38.over.chain.gz
sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp.bigChain
bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp.bigChain ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.over.bigChain

tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp.bigLink
bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp.bigLink  ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.over.bigLink
cd ..
ln -f t2tChm13_20200727.hg38.over.big* ../../../hub/t2tChm13_20200727/grch38/

** lastz all primary:
- drop non-primary and chrY from all lastz
  hg38-primary.tab

% chainFilter -q=$(cat hg38-primary.tab | tr '\n' ',') t2tChm13_20200727.hg38.all.chain.gz | bgzip --threads=64 -c >t2tChm13_20200727.hg38.all-primary.chain.gz
% cd tmp2
% hgLoadChain -noBin -test no bigChain ../t2tChm13_20200727.hg38.all-primary.chain.gz
% sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp.bigChain
% bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp.bigChain ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all-primary.bigChain

% tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp.bigLink
% bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp.bigLink  ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.all-primary.bigLink
% ln -f t2tChm13_20200727.hg38.all-primary.big* ../../../hub/t2tChm13_20200727/grch38/


and if I click on the second lastz entry I get the error
chr10:48,612,920-48,788,122
twoBitReadSeqFrag in chr10 start (133797422) >= end (46617366)

** lastz liftover primary:
% chainFilter -q=$(cat hg38-primary.tab | tr '\n' ',') t2tChm13_20200727.hg38.over.chain.gz | bgzip --threads=64 -c >t2tChm13_20200727.hg38.over-primary.chain.gz
% cd tmp3
% hgLoadChain -noBin -test no bigChain ../t2tChm13_20200727.hg38.over-primary.chain.gz
% sed 's/.000000//' chain.tab | tawk '{print $2, $4, $5, $11, 1000, $8, $3, $6, $7, $9, $10, $1}' > tmp.bigChain
% bedToBigBed -type=bed6+6 -as=${HOME}/kent/src/hg/lib/bigChain.as -tab tmp.bigChain ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.over-primary.bigChain

% tawk '{print $1, $2, $3, $5, $4}' link.tab | sort -k1,1 -k2,2n --parallel=64 > tmp.bigLink
% bedToBigBed -type=bed4+1 -as=${HOME}/kent/src/hg/lib/bigLink.as -tab tmp.bigLink  ../../genome/t2tChm13_20200727.sizes ../t2tChm13_20200727.hg38.over-primary.bigLink
% ln -f t2tChm13_20200727.hg38.over-primary.big* ../../../hub/t2tChm13_20200727/grch38/


* cenRegions
Karen Miga
team-satellite/t2t-chm13.20200727.cenRegions.bed

bgzip --threads=64 t2t-chm13.20200727.cenRegions.bed
makeMd5s t2t-chm13.20200727.cenRegions.bed.gz 
buildBigBed bed8 t2tChm13_20200727 cenRegions.bigBed t2t-chm13.20200727.cenRegions.bed.gz
ln -f cenRegions.bigBed ../../../hub/t2tChm13_20200727

* telomereRegions
/assemblies/drafts/20200727/t2t-chm13.20200727.telomere from Sergey Koren
buildBigBed bed4 t2tChm13_20200727 telomereRegions.bigBed t2t-chm13.20200727.telomere
ln -f telomereRegions.bigBed ../../../hub/t2tChm13_20200727/

* cenSat_annotation
Karen Miga
team-satellite/t2t-chm13.20200727.cenSat_annotation.bed
bgzip --threads=64 t2t-chm13.20200727.cenSat_annotation.bed 
buildBigBed bed9 t2tChm13_20200727 cenSat_annotation.bigBed t2t-chm13.20200727.cenSat_annotation.bed.gz 
ln -f cenSat_annotation.bigBed ../../../hub/t2tChm13_20200727



* GC percent 2020-09-03
cd gc5Base
hgGcPercent -noLoad -wigOut -doGaps -win=5 -file=stdout -verbose=0 NODB ../genome/t2tChm13_20200727.2bit  | gzip -c > gc5Base.wigVarStep.gz
wigToBigWig gc5Base.wigVarStep.gz  ../genome/t2tChm13_20200727.sizes gc5Base.bigWig
ln gc5Base.bigWig  ../../../hub/t2tChm13_20200727/

* TransMap all chains (refseq and gencode) 
export PATH=${HOME}/compbio/projs/transMap/transMap-dev/bin:${PATH}
cd transMap/tmp


srcDbLoadEnsemblAligns --gencodeVersion=V35 hg38 ensembl data/src/hg38/hg38.ensembl.src.db
srcDbLoadEnsemblMetadata --gencodeVersion=V35 hg38 ensembl data/src/hg38/hg38.ensembl.src.db
srcDbLoadEnsemblSeqs hg38 ensembl data/src/hg38/hg38.ensembl.src.db  

srcDbLoadGenbankAligns hg38 refseq data/src/hg38/hg38.refseq.src.db
srcDbLoadGenbankMetadata hg38 refseq data/src/hg38/hg38.refseq.src.db
srcDbLoadGenbankSeqs hg38 refseq data/src/hg38/hg38.refseq.src.db  

# manually create
sqlite3 data/genome.db < ../../../../T2T-CHM13-hub/build/t2tChm13_20200727/transMapGenomeDb.sql 

# build with all chains
#  CLUSTER DOWN DUE TO FIRE, paraHost=hgwdev
ln -s  ../../../../T2T-CHM13-hub/build/t2tChm13_20200727/transMapBuildConf.py .
mappingChainBuild data/genome.db hg38 t2tChm13_20200727 all data/chains/hg38/t2tChm13_20200727/hg38.t2tChm13_20200727.all.chain
transMapBatch ./transMapBuildConf.py hg38 t2tChm13_20200727 ensembl all >&ens.log &
transMapBatch ./transMapBuildConf.py hg38 t2tChm13_20200727 refseq all >&ref.log &

transMapBuildBigPsl ./transMapBuildConf.py t2tChm13_20200727 ensembl hg38 >&ens.log &
transMapBuildBigPsl ./transMapBuildConf.py t2tChm13_20200727 refseq hg38 >&ref.log &

cd ..
ln tmp/results/mapped/t2tChm13_20200727/t2tChm13_20200727.ensembl.transMapV1.bigPsl t2tChm13_20200727.ensembl.transMap.all.bigPsl 
ln tmp/results/mapped/t2tChm13_20200727/t2tChm13_20200727.refseq.transMapV1.bigPsl  t2tChm13_20200727.refseq.transMap.all.bigPsl 
mkdir ../../../hub/t2tChm13_20200727/transMap
ln t2tChm13_20200727.*.all.bigPsl ../../../hub/t2tChm13_20200727/transMap/

## maybe add trixIdx to make case-insensitive: ixIxx input.txt myTrix.ix myTrix.ixx

* shortRnaSeqIntrons:
cd shortRnaSeq
see t2t-chm13-gene-analysis/doc/t2tChm13_20200727.org for details
on how it was built.

buildBigBed bed12 t2tChm13_20200727 shortRnaSeqIntrons.bigBed ../../../gene-annotation/build/t2tChm13_20200727/shortRnaSeq/raw-introns-combined/shortRnaSeq-2020-08-28.juncs.bed.gz 
ln shortRnaSeqIntrons.bigBed ../../../hub/t2tChm13_20200727/

* refSeqBlat
cd refSeqBlat/data
# files from 2020-05-29
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_knownrefseq_alignments.bam
wget -nv https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.gbff.gz
faSize -detailed GRCh38_latest_rna.fna.gz | tawk '/^N/{print $1}' >refSeq-curated.acc
faSomeRecords GRCh38_latest_rna.fna.gz refSeq-curated.acc stdout | bgzip -@ 64 -c >refSeq-curated.fa.gz

# metadata for later merge (2m16s)
gbffToTsv data/GRCh38_latest_rna.gbff.gz data/refSeq-curated.meta.tsv.gz >&log&

cd ..
blatRnaMkJobs ../genome/t2tChm13_20200727.2bit data/refSeq-curated.fa.gz tmp/
para make -batch=tmp/b1 tmp/para.jobs >&log&

# filter 
faPolyASizes data/GRCh38_latest_rna.fna.gz stdout | gzip -c >data/GRCh38_latest_rna.polya.gz

find tmp/psls -type f | xargs cat | sort -k 15,15 | pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -localNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=refSeq-curated.stats stdin stdout | sort -k 14,14 -k 16,16n |gzip -9c >refSeq-curated.psl.gz

# build bigPsl with extra columns
zcat data/refSeq-curated.meta.tsv.gz | tawk '$6!=""{print $1,$6}' > tmp/rs.cds
pslToBigPsl -cds=tmp/rs.cds -fa=data/GRCh38_latest_rna.fna.gz refSeq-curated.psl.gz stdout | sort -k1,1 -k2,2n > tmp/rs.bigin1
bigBlatPslAddColumns data/refSeq-curated.meta.tsv.gz tmp/rs.bigin1 tmp/rs.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/rs.bigin2 ../genome/t2tChm13_20200727.sizes refSeq-curated.bigBlatPsl
ln refSeq-curated.bigBlatPsl  ../../../hub/t2tChm13_20200727/

* gencodeBlat
- use gencode V33 to match CAT, only do protein code and lncRNAs

 /hive/data/genomes/hg38/bed/gencodeV33/data/release_33/gencode.v33.lncRNA_transcripts.fa.gz /hive/data/genomes/hg38/bed/gencodeV33/data/release_33/gencode.v33.pc_transcripts.fa.gz  >tmp/rna.fa.gz
blatRnaMkJobs ../genome/t2tChm13_20200727.2bit tmp/rna.fa.gz tmp

# run in parasol

find tmp/psls -type f | xargs cat | sort -k 15,15 | pslCDnaFilter -minQSize=20 -ignoreIntrons -repsAsMatch -ignoreNs -bestOverlap -localNearBest=0.001  -minId=0.92 -minCover=0.75 -statsOut=gencode.stats stdin stdout | sort -k 14,14 -k 16,16n |gzip -9c >gencode.psl.gz

hgsql hg38 -Ne 'select * from wgEncodeGencodeCompV33' | cut -f 2- |genePredToFakePsl no /dev/stdin -chromSize=/hive/data/genomes/hg38/chrom.sizes /dev/null data/gencode.cds
hgsql hg38 -e 'select ga.transcriptId id, ga.geneName as geneSym, ga.geneId, gs.geneId as hgncId, ga.geneType from wgEncodeGencodeAttrsV33 ga left join wgEncodeGencodeGeneSymbolV33 gs on gs.transcriptId = ga.transcriptId' |gzip -c >data/gencode.meta.tsv.gz

# forgot to hack fasta file ids like ENST00000669853.1|ENSG00000260923.7|OTTHUMG00000172851.14|OTTHUMT00000505564.1
# also need to hack _PAR_Y ids
zcat gencode.psl.gz |tawk '$10~/_PAR_Y/{next} {$10=gensub("\\|.*$","", 1, $10);print $0}' >gencode.psl
gzip gencode.psl
zcat tmp/rna.fa.gz |awk '{$0=gensub("\\|.*$","", 1, $0);print $0}' >tmp/rna-fixed.fa

pslToBigPsl -cds=data/gencode.cds -fa=tmp/rna-fixed.fa gencode.psl.gz stdout | sort -k1,1 -k2,2n > tmp/g.bigin1
bigBlatPslAddColumns data/gencode.meta.tsv.gz tmp/g.bigin1 tmp/g.bigin2

bedToBigBed -type=bed12+17 -tab -as=../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/g.bigin2 ../genome/t2tChm13_20200727.sizes gencode.bigBlatPsl
ln gencode.bigBlatPsl  ../../../hub/t2tChm13_20200727/

* synteny  2020-09-04
cd synteny/tmp
halStats --bedSequences t2tChm13_20200727../../../data/primates_eichler.hal > chm13.bed

create syn.tmpl:
#LOOP
../../../../T2T-CHM13-hub/bin/halSyntenyRun ../../CAT/assemblyHub/t2tChm13.hal t2tChm13_20200727 $(path1) GRCh38 25000 25000 {check out exists out/25kb.$(path1).psl}
../../../../T2T-CHM13-hub/bin/halSyntenyRun ../../CAT/assemblyHub/t2tChm13.hal t2tChm13_20200727 $(path1) GRCh38 1000000 50000 {check out exists out/1mb.$(path1).psl}
#ENDLOOP

gensub2 <(cut -f 1 chm13.bed) <(echo "") syn.tmpl syn.jobs

 # run with -ram=16g
 para create -batch=b1 syn.jobs -ram=16g -maxQueue=100000000

# halSyntenty produced no output for chrMT, which is identical

# combine data for tracks
cat  out/25kb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >../synteny.25kb.psl.gz
cat  out/1mb.chr* | pslSwap stdin stdout | sort -k 14,14 -k 16,16n |gzip -9 >../synteny.1mb.psl.gz

pslToBigPsl ../synteny.25kb.psl.gz 25kb.bigin
pslToBigPsl ../synteny.1mb.psl.gz 1mb.bigin
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 25kb.bigin ../../genome/t2tChm13_20200727.sizes ../synteny.25kb.bigPsl
bedToBigBed -type=bed12+13 -tab -as=${HOME}/kent/src/hg/lib/bigPsl.as 1mb.bigin ../../genome/t2tChm13_20200727.sizes ../synteny.1mb.bigPsl

ln *.bigPsl ../../../hub/t2tChm13_20200727/synteny/


* CAT annotations 2020-09-03
# run by Marina in AWS and downloaded it to: 
   courtyard.gi.ucsc.edu:/nanopore/marina/out-chm13-t2t/assemblyHub
   source is gencode_v33


cd CAT
rsync -a courtyard.gi.ucsc.edu:/nanopore/marina/out-chm13-t2t/assemblyHub . >&log&

# need to create a GRCh38/CAT and add to GRCh38 to assembly hub
ln assemblyHub/GRCh38/annotation.bb assemblyHub/GRCh38/GRCh38.2bit  ../../../hub/GRCh38/

ln assemblyHub/CHM13/*.bam ../../../hub/t2tChm13_20200727/CAT/
ln assemblyHub/CHM13/*.bam.bai ../../../hub/t2tChm13_20200727/CAT/
ln assemblyHub/CHM13/*.bb   ../../../hub/t2tChm13_20200727/CAT/

# rename HAL
export PATH=${HOME}/compbio/compartiveGenomics/projs/hal/src/hal/bin:${PATH}
CHM13	t2tChm13_20200727
halRenameGenomes assemblyHub/t2tChm13.hal rename.tab
halStats --genomes assemblyHub/t2tChm13.hal
ln assemblyHub/t2tChm13.hal   ../../../hub/t2tChm13_20200727/CAT/

# make combined BED of all reads to make it posible to
# see context
# whack read names to make it smaller
cnvBedToBam:
  #!/bin/bash -e
  bamToBed -bed12 -i $1 | tawk '{$4="";print}' > $2

find assemblyHub/CHM13 -name '*.bam' | parallel -j 8 ./cnvBamToBed {} tmp/beds/{/}.bed&
sort -k1,1 -k2,2n -m --parallel=64 tmp/beds/*.bed >tmp/all_isoseq.bed
buildBigBed bed12 t2tChm13_20200727 all_isoseq.bigBed tmp/all_isoseq.bed
ln all_isoseq.bigBed  ../../../hub/t2tChm13_20200727/CAT/

# 2020-09-14 update in bb and databases from
http://courtyard.gi.ucsc.edu/~mhauknes/t2t/out-chm13-t2t-v5/assemblyHub/hub.txt
ln -f assemblyHub/CHM13/*.bb   ../../../hub/t2tChm13_20200727/CAT/

*** WARNING: we found that CAT hand included user-filtered PBs, recreate
cd CAT/consensus_gene_set
tawk '$14 != "user_filtered"' CHM13.gp_info >CHM13.no-user_filtered.gp_info
selectById 2 CHM13.no-user_filterd.gp_info 1 CHM13.gp  > CHM13.no-user_filtered.gp

cd CAT/assemblyHub/CHM13/
mv consensus.bb  consensus.orig.bb 
bigBedToBed consensus.orig.bb consensus.orig.bed
# 37 columns 32 has user_filtered
tawk '$32 != "user_filtered" ' consensus.orig.bed > consensus.no-user_filtered.bed
buildBigBed --as=cat.as bed12+20 t2tChm13_20200727 consensus.no-user_filtered.bb consensus.no-user_filtered.bed 



* restriction enzyme file:
- don't build as a track, as this is done on the fly in the browser, but build for Karen for analysis
cd rebaseData
curl -A \"Mozilla/4.0\" http://rebase.neb.com/rebase/link_gcgenz > rebase.gcg
# need to build findCutters.c
cut -f 1 ../genome/t2tChm13_20200727.sizes | parallel findCutters rebase.gcg ../genome/t2tChm13_20200727.2bit:{} tmp/{}.bed
# NOTE: can't merge since findCutters output is not sorted :-(
sort -T /dev/shm --parallel=64 -k 1,1 -k 2,2n tmp/chr*.bed |bgzip --threads=64 -c >rebase.bed.gz &
tabix -p bed rebase.bed.gz 

* uniqRegions
attempts to find regions not in GRCh38
** with lastz over chains
# requires 3-column sizes
tawk '{print $1,$2,"x"}' ../genome/t2tChm13_20200727.sizes >lastz/t2tChm13_20200727.sizes3

featureBits -not -bed=lastz/lastzUnalignedOver.bed -countGaps -chromSize=lastz/t2tChm13_20200727.sizes3 nodb ../hg38Lastz/t2tChm13_20200727.hg38.over.chain.gz
gzip lastz/lastzUnalignedOver.bed 
buildBigBed bed4 t2tChm13_20200727 lastz/lastzUnalignedOver.bigBed lastz/lastzUnalignedOver.bed.gz 
ln -f lastz/lastzUnalignedOver.bigBed  ../../../hub/t2tChm13_20200727/uniqRegions/

** with primary assembly lastz over chains
# requires 3-column sizes
featureBits -not -bed=lastz/lastzUnalignedOverPrimary.bed -countGaps -chromSize=lastz/t2tChm13_20200727.sizes3 nodb ../hg38Lastz/t2tChm13_20200727.hg38.over-primary.chain.gz
gzip lastz/lastzUnalignedOverPrimary.bed 
buildBigBed bed4 t2tChm13_20200727 lastz/lastzUnalignedOverPrimary.bigBed lastz/lastzUnalignedOverPrimary.bed.gz 
ln lastz/lastzUnalignedOverPrimary.bigBed  ../../../hub/t2tChm13_20200727/uniqRegions/


** minimap chrUn and random now in primary assembly
cd minimap-unplaced
tawk '/Un/||/random/{print $1}' /hive/data/genomes/hg38/chrom.sizes | sort > unplaced-random.chroms
tawk '/Un/||/random/{print $1,$2}' /hive/data/genomes/hg38/chrom.sizes | sort> unplaced-random.sizes

cd tmp
twoBitToFa -seqList=../unplaced-random.chroms /hive/data/genomes/hg38/hg38.2bit hg38.unplaced.fa
zcat ../../../genome/t2tChm13_20200727.fa.gz >t2tChm13_20200727.fa
minimap2 -x asm10 -d t2tChm13_20200727.mmi t2tChm13_20200727.fa
minimap2 -ax asm10 t2tChm13_20200727.fa hg38.unplaced.fa > hg38-unplaced-chm13.sam 

samtools sort hg38-unplaced-chm13.sam | samtools view -b  >hg38-unplaced-chm13.bam
samtools index hg38-unplaced-chm13.bam 
cd ..
bamToBed -bed12 -i hg38-unplaced-chm13.bam >hg38-unplaced-chm13.bed

buildBigBed bed12 t2tChm13_20200727 hg38-unplaced-chm13.bigBed hg38-unplaced-chm13.bed

cut -f 4 hg38-unplaced-chm13.bed | sort -u > placed-unplaced.chroms
selectById 1 placed-unplaced.chroms 1 unplaced-random.sizes > placed-unplaced.sizes
selectById -not 1 placed-unplaced.sizes 1 unplaced-random.sizes >still-unplaced.sizes

down to two unplaced, and we don't have chrY
  chrUn_KI270752v1    27745
  chrY_KI270740v1_random 37240

ln -f hg38-unplaced-chm13.bigBed  ../../../../hub/t2tChm13_20200727/uniqRegions/

** cactus chrUn/random and unique
cd uniqRegions/cactus

halStats --bedSequences t2tChm13_20200727 ../../CAT/assemblyHub/t2tChm13.hal  > tmp/chm13.bed
tawk '{print $0 >"tmp/"$1".chrom.bed"}' tmp/chm13.bed 
ls -1 tmp/*.chrom.bed |nice parallel halLiftover --outPSL ../../CAT/assemblyHub/t2tChm13.hal t2tChm13_20200727 {} GRCh38 {}.psl
# opps, did this backwards, with CHM13 being query
find tmp -name '*.psl' | xargs cat | pslSwap stdin stdout | pslPosTarget stdin stdout | sort --parallel=64 -k14,14 -k16,16n | bgzip --threads=64 >cactus-pairwise.psl.gz

# inverse
featureBits -not -bed=cactus-unique.bed -countGaps -chromSize=../lastz/t2tChm13_20200727.sizes3 nodb cactus-pairwise.psl.gz &
buildBigBed bed4 t2tChm13_20200727 cactus-unique.bigBed cactus-unique.bed 

# inverse primary
zcat cactus-pairwise.psl.gz |tawk '($10!~/_/) && ($10!="chrY")'  |  bgzip --threads=64 >  cactus-pairwise.primary.psl.gz
featureBits -not -bed=cactus-unique-primary.bed -countGaps -chromSize=../lastz/t2tChm13_20200727.sizes3 nodb cactus-pairwise.primary.psl.gz 
buildBigBed bed4 t2tChm13_20200727 cactus-unique-primary.bigBed cactus-unique-primary.bed 

# unplaced
zcat cactus-pairwise.psl.gz | tawk '$10 ~ /Un|random/' > cactus-unplaced.psl
pslToBed cactus-unplaced.psl cactus-unplaced.bed 
buildBigBed bed12 t2tChm13_20200727 cactus-unplaced.bigBed cactus-unplaced.bed 

ln -f *.bigBed  ../../../../hub/t2tChm13_20200727/uniqRegions

* proseq
2020-09-08 Savannah Klein
/team-epigenetics/20200727_PROseq_CHM13
% wigToBigWig CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-100mm-0m-chm13-v1.0_Aligned.sortedByCoord_NEG.bedgraph.gz ../genome/t2tChm13_20200727.sizes proSeq-neg.bigWig &
% wigToBigWig CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-100mm-0m-chm13-v1.0_Aligned.sortedByCoord_POS.bedgraph.gz ../genome/t2tChm13_20200727.sizes proSeq-pos.bigWig &
% ln  proSeq-* ../../../hub/t2tChm13_20200727/proSeq

* liftOffGenes
Mitchell Robert Vollger  
t2t_globus_share/team-segdups/Assembly_analysis/Liftoff/bigGenePred.as
t2t_globus_share/team-segdups/Assembly_analysis/Liftoff/liftoff.html
t2t_globus_share/team-segdups/Assembly_analysis/Liftoff/t2t-chm13.20200727.all.bed
t2t_globus_share/team-segdups/Assembly_analysis/Liftoff/t2t-chm13.20200727.orf_only.bed

% bedToBigBed -type=bed12+8 -tab -as=/cluster/home/markd/kent/src/hg/lib/bigGenePred.as t2t-chm13.20200727.all.bed ../genome/t2tChm13_20200727.sizes liftOffGenes.all.bigGenePred
% bedToBigBed -type=bed12+8 -tab -as=/cluster/home/markd/kent/src/hg/lib/bigGenePred.as t2t-chm13.20200727.orf_only.bed ../genome/t2tChm13_20200727.sizes liftOffGenes.orf_only.bigGenePred

% ln *.bigGenePred  ../../../hub/t2tChm13_20200727/liftOffGenes/


* fosmidABC10
Mitchell Robert Vollger
team-segdups/FES_alignments/*
ln -f *.bb   ../../../hub/t2tChm13_20200727/fosmidABC10/


* hg002Aligns 
/team-variants/assembly_aligns/for_browser_hub/t2tChm13_20200727/HG002_hifiasm_assemblies
Nancy Hansen
makeMd5s *.bam *.bam.bai&
ln -f *.bam *.bai ../../../hub/t2tChm13_20200727/hg002Aligns/

o  11:55 AM
Hi again, Mark! I've checked over how the nucmer bam files look in the browser (and in fact, found a minor bug in how my delta->sam parser constructs cigar strings, so thanks!) and I've now put minimap2 bams for the HG002 assemblies in the Globus directory as well. Can you load the contents of team-variants/assembly_aligns/for_browser_hub/t2tChm13_20200727/HG002_hifiasm_assemblies into the t2tChm13_20200727 part of the hub? I created a file trackDb.hg002_assemblies.txt with what I think would be the entries in your trackDb.txt, but of course I may have gotten it wrong, so it's probably worth a quick check.


* todo
** RMMaskBed to David, etc
** as_sf_annotation.html as_hor_annotation.html reference a google doc
** chains for hg38->t2t, announce to variant group.


** introns /team-genes/20200727/shortRnaSeq/raw-introns
** GENCODE and RefSeq BLAT alignments
** mydennis  CN mapps
We generated CN maps of the 20200727 release using kmers and CHM13 genome Illumina data (quickmer2 pipeline). Not sure if this would be of interest to add as a track
track name=CHM13_CN itemRgb="On" bigDataUrl=https://bioshare.bioinformatics.ucdavis.edu/bioshare/download/rlc692m7tk5cibb/t2t/t2t_chm13_20200727/quickmer2/SRR3986881.depth.bed.CN.bb type=bigBed
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7073954/
** proseq  (@Savannah Klein @Rachel O'Neill) 
**  RNA-seq PE ILMN mydennis
** all isoset 
** check default assembly in CHM13
** report hide bug

** lastzall all-primary bug
chr10:48,612,920-48,788,122
and if I click on the second lastz entry I get the error
twoBitReadSeqFrag in chr10 start (133797422) >= end (46617366)

converting to psl and doing a pslCheck shows alignment is fine.



* next assembly
** minimap grch38 with chaining (but this missed chr22:1,093-476,524 dup case)
** cactus pairwise for unplaced regions
