<h2>Description</h2>
<p>Merqury - Assembly-only (Errors)</p>


<h2>Methods</h2>
<p>The first 23 bp of the 10X Genomics read pair 1 were trimmed to remove barcode bases,
and 21-mers were collected with Meryl (read set).


Merqury was run on t2t-chm13.20200602.fasta with the read set, collecting assembly-only (error) k-mers.


Each position containing an error k-mer was converted to a .bed file, with the accumulated number of k-mers per position collected in bigwig format. As a result, the maximum number of k-mers found per base will be 21 (all k-mers overlapping the base are errors).
</p>


<h2>Display Conventions and Configuration</h2>
<!-- if there are special conventions, like colors, describe them here -->
<p>Maximum set to 21x</p>


<h2>Data access</h2>
<!-- Optional, links for downloading the data if there is a separate source beside the
  browser  -->
<p>Data is available on globus: team-curation/merqury/20200602.only.bigWig with corresponding 20200602.chr_only.tdf and 20200602.nonChr_only.tdf</p>


<h2>Release history</h2>
<ol>
  <li> 2020 June 9 : Initial upload for 20200602 release
</ol>


<h2>Contact</h2>
<p>Contact <a href="mailto:rhiea@nih.gov">Arang Rhie &lt;rhiea@nih.gov&gt;</a>
</p>


<h2>Credits</h2>
<p></p>
